{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "source": [
        "#instal part"
      ],
      "metadata": {
        "id": "Pmikxj-ZVKp_"
      }
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "PMcRTU1nUAvn",
        "outputId": "3626c5c1-b48f-432e-be3e-654d9e5ea8d6"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2025-05-27 06:36:09--  https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh\n",
            "Resolving repo.anaconda.com (repo.anaconda.com)... 104.16.32.241, 104.16.191.158, 2606:4700::6810:20f1, ...\n",
            "Connecting to repo.anaconda.com (repo.anaconda.com)|104.16.32.241|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 155472915 (148M) [application/octet-stream]\n",
            "Saving to: ‘Miniconda3-latest-Linux-x86_64.sh’\n",
            "\n",
            "Miniconda3-latest-L 100%[===================>] 148.27M   255MB/s    in 0.6s    \n",
            "\n",
            "2025-05-27 06:36:10 (255 MB/s) - ‘Miniconda3-latest-Linux-x86_64.sh’ saved [155472915/155472915]\n",
            "\n",
            "PREFIX=/usr/local\n",
            "Unpacking payload ...\n",
            "entry_point.py:256: DeprecationWarning: Python 3.14 will, by default, filter extracted tar archives and reject files or modify their metadata. Use the filter argument to control this behavior.\n",
            "entry_point.py:256: DeprecationWarning: Python 3.14 will, by default, filter extracted tar archives and reject files or modify their metadata. Use the filter argument to control this behavior.\n",
            "\n",
            "Installing base environment...\n",
            "\n",
            "Preparing transaction: ...working... done\n",
            "Executing transaction: ...working... done\n",
            "entry_point.py:256: DeprecationWarning: Python 3.14 will, by default, filter extracted tar archives and reject files or modify their metadata. Use the filter argument to control this behavior.\n",
            "installation finished.\n",
            "WARNING:\n",
            "    You currently have a PYTHONPATH environment variable set. This may cause\n",
            "    unexpected behavior when running the Python interpreter in Miniconda3.\n",
            "    For best results, please verify that your PYTHONPATH only points to\n",
            "    directories of packages that are compatible with the Python interpreter\n",
            "    in Miniconda3: /usr/local\n"
          ]
        }
      ],
      "source": [
        "!wget -c https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh\n",
        "!chmod +x Miniconda3-latest-Linux-x86_64.sh\n",
        "!bash ./Miniconda3-latest-Linux-x86_64.sh -b -f -p /usr/local\n",
        "import sys\n",
        "sys.path.append('/usr/local/lib/python3.8/site-packages')\n"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!conda init bash\n",
        "!source ~/.bashrc"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "vGAt8zNQU0Q_",
        "outputId": "c07f6edf-bef6-4a05-e312-95a5f7a3f13c"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "no change     /usr/local/condabin/conda\n",
            "no change     /usr/local/bin/conda\n",
            "no change     /usr/local/bin/conda-env\n",
            "no change     /usr/local/bin/activate\n",
            "no change     /usr/local/bin/deactivate\n",
            "no change     /usr/local/etc/profile.d/conda.sh\n",
            "no change     /usr/local/etc/fish/conf.d/conda.fish\n",
            "no change     /usr/local/shell/condabin/Conda.psm1\n",
            "no change     /usr/local/shell/condabin/conda-hook.ps1\n",
            "no change     /usr/local/lib/python3.13/site-packages/xontrib/conda.xsh\n",
            "no change     /usr/local/etc/profile.d/conda.csh\n",
            "modified      /root/.bashrc\n",
            "\n",
            "==> For changes to take effect, close and re-open your current shell. <==\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!conda install -c bioconda hisat2"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "FHenfRP9Ay1j",
        "outputId": "175c8694-47fc-4520-c9f7-626a91c2c1f9"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Channels:\n",
            " - bioconda\n",
            " - defaults\n",
            "Platform: linux-64\n",
            "Collecting package metadata (repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "Solving environment: \\ \b\b| \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - hisat2\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    hisat2-2.2.1               |       h87f3376_4        16.5 MB  bioconda\n",
            "    perl-5.40.2                | 0_h5eee18b_perl5        13.2 MB\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:        29.7 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  hisat2             bioconda/linux-64::hisat2-2.2.1-h87f3376_4 \n",
            "  perl               pkgs/main/linux-64::perl-5.40.2-0_h5eee18b_perl5 \n",
            "\n",
            "\n",
            "Proceed ([y]/n)? y\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages:\n",
            "hisat2-2.2.1         | 16.5 MB   | :   0% 0/1 [00:00<?, ?it/s]\n",
            "hisat2-2.2.1         | 16.5 MB   | :  29% 0.29195923204025953/1 [00:00<00:00,  2.92it/s]\n",
            "perl-5.40.2          | 13.2 MB   | :  49% 0.48903687677464525/1 [00:00<00:00,  4.89it/s]\u001b[A\n",
            "perl-5.40.2          | 13.2 MB   | : 100% 1.0/1 [00:00<00:00,  3.77it/s]                \u001b[A\n",
            "hisat2-2.2.1         | 16.5 MB   | : 100% 1.0/1 [00:00<00:00,  3.09it/s]\n",
            "                                                                        \n",
            "                                                                        \u001b[A\n",
            "Preparing transaction: - \b\bdone\n",
            "Verifying transaction: | \b\b/ \b\bdone\n",
            "Executing transaction: \\ \b\b| \b\bdone\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!conda install -c conda-forge -c bioconda -y snakemake\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "jo9Fviw2U9Mv",
        "outputId": "d023669e-fd8f-464d-f52f-e43f66619ca5"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Channels:\n",
            " - conda-forge\n",
            " - bioconda\n",
            " - defaults\n",
            "Platform: linux-64\n",
            "Collecting package metadata (repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "Solving environment: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bfailed\n",
            "\n",
            "LibMambaUnsatisfiableError: Encountered problems while solving:\n",
            "  - nothing provides snakemake-minimal 8.11.2.* needed by snakemake-8.11.2-hdfd78af_0\n",
            "\n",
            "Could not solve for environment specs\n",
            "The following packages are incompatible\n",
            "├─ \u001b[32mpin on python 3.13.* =* *\u001b[0m is installable and it requires\n",
            "│  └─ \u001b[32mpython =3.13 *\u001b[0m, which can be installed;\n",
            "└─ \u001b[31msnakemake =* *\u001b[0m is not installable because there are no viable options\n",
            "   ├─ \u001b[31msnakemake [3.10.0|3.10.1|...|3.9.1]\u001b[0m would require\n",
            "   │  └─ \u001b[31mpython =3.4 *\u001b[0m, which conflicts with any installable versions previously reported;\n",
            "   ├─ \u001b[31msnakemake [3.10.0|3.10.1|...|5.3.0]\u001b[0m would require\n",
            "   │  └─ \u001b[31mpython [=3.5 *|>=3.5,<3.6.0a0 *]\u001b[0m, which conflicts with any installable versions previously reported;\n",
            "   ├─ \u001b[31msnakemake [3.11.2|3.12.0|...|5.1.4]\u001b[0m would require\n",
            "   │  └─ \u001b[31mpython =3.6 *\u001b[0m, which conflicts with any installable versions previously reported;\n",
            "   ├─ \u001b[31msnakemake [5.1.4|5.1.5|...|5.3.0]\u001b[0m would require\n",
            "   │  └─ \u001b[31mpython >=3.6,<3.7.0a0 *\u001b[0m, which conflicts with any installable versions previously reported;\n",
            "   ├─ \u001b[31msnakemake 5.2.1\u001b[0m would require\n",
            "   │  └─ \u001b[31msnakemake-minimal =5.2.1 *\u001b[0m but there are no viable options\n",
            "   │     ├─ \u001b[31msnakemake-minimal [5.2.1|5.3.0]\u001b[0m would require\n",
            "   │     │  └─ \u001b[31mpython >=3.5,<3.6.0a0 *\u001b[0m, which conflicts with any installable versions previously reported;\n",
            "   │     └─ \u001b[31msnakemake-minimal [5.2.1|5.3.0]\u001b[0m would require\n",
            "   │        └─ \u001b[31mpython >=3.6,<3.7.0a0 *\u001b[0m, which conflicts with any installable versions previously reported;\n",
            "   ├─ \u001b[31msnakemake [5.10.0|5.11.0|...|7.9.0]\u001b[0m would require\n",
            "   │  └─ \u001b[31msnakemake-minimal [=5.10.0 *|=5.11.0 *|...|=7.9.0 *]\u001b[0m but there are no viable options\n",
            "   │     ├─ \u001b[31msnakemake-minimal [5.10.0|5.11.0|...|7.9.0]\u001b[0m would require\n",
            "   │     │  └─ \u001b[31mratelimiter =* *\u001b[0m but there are no viable options\n",
            "   │     │     ├─ \u001b[31mratelimiter 1.2.0\u001b[0m would require\n",
            "   │     │     │  └─ \u001b[31mpython [=2.7 *|>=2.7,<2.8.0a0 *]\u001b[0m, which conflicts with any installable versions previously reported;\n",
            "   │     │     ├─ \u001b[31mratelimiter 1.2.0\u001b[0m would require\n",
            "   │     │     │  └─ \u001b[31mpython [=3.5 *|>=3.5,<3.6.0a0 *]\u001b[0m, which conflicts with any installable versions previously reported;\n",
            "   │     │     ├─ \u001b[31mratelimiter 1.2.0\u001b[0m would require\n",
            "   │     │     │  └─ \u001b[31mpython =3.6 *\u001b[0m, which conflicts with any installable versions previously reported;\n",
            "   │     │     ├─ \u001b[31mratelimiter 1.2.0\u001b[0m would require\n",
            "   │     │     │  └─ \u001b[31mpython >=3.6,<3.7.0a0 *\u001b[0m, which conflicts with any installable versions previously reported;\n",
            "   │     │     ├─ \u001b[31mratelimiter 1.2.0\u001b[0m would require\n",
            "   │     │     │  └─ \u001b[31mpython >=3.7,<3.8.0a0 *\u001b[0m, which conflicts with any installable versions previously reported;\n",
            "   │     │     ├─ \u001b[31mratelimiter 1.2.0\u001b[0m would require\n",
            "   │     │     │  └─ \u001b[31mpython >=3.8,<3.9.0a0 *\u001b[0m, which conflicts with any installable versions previously reported;\n",
            "   │     │     └─ \u001b[31mratelimiter 1.2.0\u001b[0m would require\n",
            "   │     │        └─ \u001b[31mpython >=3,<3.11 *\u001b[0m, which conflicts with any installable versions previously reported;\n",
            "   │     ├─ \u001b[31msnakemake-minimal [5.2.1|5.3.0]\u001b[0m, which cannot be installed (as previously explained);\n",
            "   │     └─ \u001b[31msnakemake-minimal [5.2.1|5.3.0]\u001b[0m, which cannot be installed (as previously explained);\n",
            "   ├─ \u001b[31msnakemake 5.23.0\u001b[0m would require\n",
            "   │  └─ \u001b[31mpulp =* *\u001b[0m but there are no viable options\n",
            "   │     ├─ \u001b[31mpulp [2.5.1|2.6.0|2.7.0|2.8.0]\u001b[0m would require\n",
            "   │     │  └─ \u001b[31mpython >=3.10,<3.11.0a0 *\u001b[0m, which conflicts with any installable versions previously reported;\n",
            "   │     ├─ \u001b[31mpulp [2.6.0|2.7.0|2.8.0]\u001b[0m would require\n",
            "   │     │  └─ \u001b[31mpython >=3.11,<3.12.0a0 *\u001b[0m, which conflicts with any installable versions previously reported;\n",
            "   │     ├─ \u001b[31mpulp [2.7.0|2.8.0]\u001b[0m would require\n",
            "   │     │  └─ \u001b[31mpython [>=3.12,<3.13.0a0 *|>=3.12.0rc3,<3.13.0a0 *]\u001b[0m, which conflicts with any installable versions previously reported;\n",
            "   │     ├─ \u001b[31mpulp [2.3|2.3.1|...|2.8.0]\u001b[0m would require\n",
            "   │     │  └─ \u001b[31mpython >=3.8,<3.9.0a0 *\u001b[0m, which conflicts with any installable versions previously reported;\n",
            "   │     ├─ \u001b[31mpulp [2.3|2.3.1|...|2.8.0]\u001b[0m would require\n",
            "   │     │  └─ \u001b[31mpython >=3.9,<3.10.0a0 *\u001b[0m, which conflicts with any installable versions previously reported;\n",
            "   │     ├─ \u001b[31mpulp 1.6.8\u001b[0m would require\n",
            "   │     │  └─ \u001b[31mpython [=2.7 *|>=2.7,<2.8.0a0 *]\u001b[0m, which conflicts with any installable versions previously reported;\n",
            "   │     ├─ \u001b[31mpulp 1.6.8\u001b[0m would require\n",
            "   │     │  └─ \u001b[31mpython =3.5 *\u001b[0m, which conflicts with any installable versions previously reported;\n",
            "   │     ├─ \u001b[31mpulp 1.6.8\u001b[0m would require\n",
            "   │     │  └─ \u001b[31mpython =3.6 *\u001b[0m, which conflicts with any installable versions previously reported;\n",
            "   │     ├─ \u001b[31mpulp [1.6.8|2.3|...|2.5.1]\u001b[0m would require\n",
            "   │     │  └─ \u001b[31mpython >=3.6,<3.7.0a0 *\u001b[0m, which conflicts with any installable versions previously reported;\n",
            "   │     └─ \u001b[31mpulp [1.6.8|2.3|...|2.6.0]\u001b[0m would require\n",
            "   │        └─ \u001b[31mpython >=3.7,<3.8.0a0 *\u001b[0m, which conflicts with any installable versions previously reported;\n",
            "   ├─ \u001b[31msnakemake 7.18.2\u001b[0m would require\n",
            "   │  └─ \u001b[31msnakemake-minimal =7.18.2 *\u001b[0m but there are no viable options\n",
            "   │     ├─ \u001b[31msnakemake-minimal [7.18.2|7.19.0|...|7.32.4]\u001b[0m would require\n",
            "   │     │  └─ \u001b[31mpulp >=2.0,<2.8.0 *\u001b[0m, which cannot be installed (as previously explained);\n",
            "   │     └─ \u001b[31msnakemake-minimal [5.10.0|5.11.0|...|7.9.0]\u001b[0m, which cannot be installed (as previously explained);\n",
            "   ├─ \u001b[31msnakemake [7.19.0|7.19.1|...|7.32.4]\u001b[0m would require\n",
            "   │  └─ \u001b[31msnakemake-minimal [=7.19.0 *|=7.19.1 *|...|=7.32.4 *]\u001b[0m but there are no viable options\n",
            "   │     ├─ \u001b[31msnakemake-minimal [7.18.2|7.19.0|...|7.32.4]\u001b[0m, which cannot be installed (as previously explained);\n",
            "   │     └─ \u001b[31msnakemake-minimal 7.32.4\u001b[0m would require\n",
            "   │        └─ \u001b[31mpython >=3.7,<3.12 *\u001b[0m, which conflicts with any installable versions previously reported;\n",
            "   ├─ \u001b[31msnakemake [8.0.0|8.0.1|...|9.5.1]\u001b[0m would require\n",
            "   │  └─ \u001b[31msnakemake-minimal [=8.0.0 *|=8.0.1 *|...|=9.5.1 *]\u001b[0m, which requires\n",
            "   │     └─ \u001b[31mpython >=3.11,<3.13 *\u001b[0m, which conflicts with any installable versions previously reported;\n",
            "   └─ \u001b[31msnakemake 8.11.2\u001b[0m would require\n",
            "      └─ \u001b[31msnakemake-minimal =8.11.2 *\u001b[0m, which does not exist (perhaps a missing channel).\n",
            "\n",
            "Pins seem to be involved in the conflict. Currently pinned specs:\n",
            " - python=3.13\n",
            "\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install snakemake"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "h41X4fNkVA2N",
        "outputId": "d9cbe59d-7046-4ac4-e99e-582d57993a27"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting snakemake\n",
            "  Downloading snakemake-9.5.1-py3-none-any.whl.metadata (2.7 kB)\n",
            "Collecting appdirs (from snakemake)\n",
            "  Downloading appdirs-1.4.4-py2.py3-none-any.whl.metadata (9.0 kB)\n",
            "Collecting immutables (from snakemake)\n",
            "  Downloading immutables-0.21-cp313-cp313-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (4.5 kB)\n",
            "Collecting configargparse (from snakemake)\n",
            "  Downloading configargparse-1.7.1-py3-none-any.whl.metadata (24 kB)\n",
            "Collecting connection_pool>=0.0.3 (from snakemake)\n",
            "  Downloading connection_pool-0.0.3.tar.gz (3.8 kB)\n",
            "  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting docutils (from snakemake)\n",
            "  Downloading docutils-0.21.2-py3-none-any.whl.metadata (2.8 kB)\n",
            "Collecting gitpython (from snakemake)\n",
            "  Downloading GitPython-3.1.44-py3-none-any.whl.metadata (13 kB)\n",
            "Collecting humanfriendly (from snakemake)\n",
            "  Downloading humanfriendly-10.0-py2.py3-none-any.whl.metadata (9.2 kB)\n",
            "Collecting jinja2<4.0,>=3.0 (from snakemake)\n",
            "  Downloading jinja2-3.1.6-py3-none-any.whl.metadata (2.9 kB)\n",
            "Collecting jsonschema (from snakemake)\n",
            "  Downloading jsonschema-4.24.0-py3-none-any.whl.metadata (7.8 kB)\n",
            "Collecting nbformat (from snakemake)\n",
            "  Downloading nbformat-5.10.4-py3-none-any.whl.metadata (3.6 kB)\n",
            "Requirement already satisfied: packaging in /usr/local/lib/python3.13/site-packages (from snakemake) (24.2)\n",
            "Collecting psutil (from snakemake)\n",
            "  Downloading psutil-7.0.0-cp36-abi3-manylinux_2_12_x86_64.manylinux2010_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (22 kB)\n",
            "Collecting pulp<3.2,>=2.3.1 (from snakemake)\n",
            "  Downloading pulp-3.1.1-py3-none-any.whl.metadata (1.3 kB)\n",
            "Collecting pyyaml (from snakemake)\n",
            "  Downloading PyYAML-6.0.2-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (2.1 kB)\n",
            "Requirement already satisfied: requests<3.0,>=2.8.1 in /usr/local/lib/python3.13/site-packages (from snakemake) (2.32.3)\n",
            "Collecting reretry (from snakemake)\n",
            "  Downloading reretry-0.11.8-py2.py3-none-any.whl.metadata (5.5 kB)\n",
            "Collecting smart-open<8.0,>=4.0 (from snakemake)\n",
            "  Downloading smart_open-7.1.0-py3-none-any.whl.metadata (24 kB)\n",
            "Collecting snakemake-interface-executor-plugins<10.0,>=9.3.2 (from snakemake)\n",
            "  Downloading snakemake_interface_executor_plugins-9.3.5-py3-none-any.whl.metadata (7.8 kB)\n",
            "Collecting snakemake-interface-common<2.0,>=1.17.0 (from snakemake)\n",
            "  Downloading snakemake_interface_common-1.18.0-py3-none-any.whl.metadata (474 bytes)\n",
            "Collecting snakemake-interface-storage-plugins<5.0,>=4.1.0 (from snakemake)\n",
            "  Downloading snakemake_interface_storage_plugins-4.2.1-py3-none-any.whl.metadata (11 kB)\n",
            "Collecting snakemake-interface-report-plugins<2.0.0,>=1.1.0 (from snakemake)\n",
            "  Downloading snakemake_interface_report_plugins-1.1.0-py3-none-any.whl.metadata (3.4 kB)\n",
            "Collecting snakemake-interface-logger-plugins<2.0.0,>=1.1.0 (from snakemake)\n",
            "  Downloading snakemake_interface_logger_plugins-1.2.3-py3-none-any.whl.metadata (859 bytes)\n",
            "Collecting tabulate (from snakemake)\n",
            "  Downloading tabulate-0.9.0-py3-none-any.whl.metadata (34 kB)\n",
            "Collecting throttler (from snakemake)\n",
            "  Downloading throttler-1.2.2-py3-none-any.whl.metadata (7.4 kB)\n",
            "Collecting wrapt (from snakemake)\n",
            "  Downloading wrapt-1.17.2-cp313-cp313-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (6.4 kB)\n",
            "Collecting yte<2.0,>=1.5.5 (from snakemake)\n",
            "  Downloading yte-1.7.0-py3-none-any.whl.metadata (3.5 kB)\n",
            "Collecting dpath<3.0.0,>=2.1.6 (from snakemake)\n",
            "  Downloading dpath-2.2.0-py3-none-any.whl.metadata (15 kB)\n",
            "Collecting conda-inject<2.0,>=1.3.1 (from snakemake)\n",
            "  Downloading conda_inject-1.3.2-py3-none-any.whl.metadata (855 bytes)\n",
            "Collecting MarkupSafe>=2.0 (from jinja2<4.0,>=3.0->snakemake)\n",
            "  Downloading MarkupSafe-3.0.2-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (4.0 kB)\n",
            "Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.13/site-packages (from requests<3.0,>=2.8.1->snakemake) (3.3.2)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.13/site-packages (from requests<3.0,>=2.8.1->snakemake) (3.7)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.13/site-packages (from requests<3.0,>=2.8.1->snakemake) (2.3.0)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.13/site-packages (from requests<3.0,>=2.8.1->snakemake) (2025.4.26)\n",
            "Collecting argparse-dataclass>=2.0.0 (from snakemake-interface-common<2.0,>=1.17.0->snakemake)\n",
            "  Downloading argparse_dataclass-2.0.0-py3-none-any.whl.metadata (7.2 kB)\n",
            "Collecting plac<2.0.0,>=1.3.4 (from yte<2.0,>=1.5.5->snakemake)\n",
            "  Downloading plac-1.4.5-py2.py3-none-any.whl.metadata (5.9 kB)\n",
            "Collecting gitdb<5,>=4.0.1 (from gitpython->snakemake)\n",
            "  Downloading gitdb-4.0.12-py3-none-any.whl.metadata (1.2 kB)\n",
            "Collecting attrs>=22.2.0 (from jsonschema->snakemake)\n",
            "  Downloading attrs-25.3.0-py3-none-any.whl.metadata (10 kB)\n",
            "Collecting jsonschema-specifications>=2023.03.6 (from jsonschema->snakemake)\n",
            "  Downloading jsonschema_specifications-2025.4.1-py3-none-any.whl.metadata (2.9 kB)\n",
            "Collecting referencing>=0.28.4 (from jsonschema->snakemake)\n",
            "  Downloading referencing-0.36.2-py3-none-any.whl.metadata (2.8 kB)\n",
            "Collecting rpds-py>=0.7.1 (from jsonschema->snakemake)\n",
            "  Downloading rpds_py-0.25.1-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (4.1 kB)\n",
            "Collecting fastjsonschema>=2.15 (from nbformat->snakemake)\n",
            "  Downloading fastjsonschema-2.21.1-py3-none-any.whl.metadata (2.2 kB)\n",
            "Collecting jupyter-core!=5.0.*,>=4.12 (from nbformat->snakemake)\n",
            "  Downloading jupyter_core-5.8.0-py3-none-any.whl.metadata (3.3 kB)\n",
            "Collecting traitlets>=5.1 (from nbformat->snakemake)\n",
            "  Downloading traitlets-5.14.3-py3-none-any.whl.metadata (10 kB)\n",
            "Collecting smmap<6,>=3.0.1 (from gitdb<5,>=4.0.1->gitpython->snakemake)\n",
            "  Downloading smmap-5.0.2-py3-none-any.whl.metadata (4.3 kB)\n",
            "Requirement already satisfied: platformdirs>=2.5 in /usr/local/lib/python3.13/site-packages (from jupyter-core!=5.0.*,>=4.12->nbformat->snakemake) (4.3.7)\n",
            "Downloading snakemake-9.5.1-py3-none-any.whl (1.1 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.1/1.1 MB\u001b[0m \u001b[31m50.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading conda_inject-1.3.2-py3-none-any.whl (4.1 kB)\n",
            "Downloading dpath-2.2.0-py3-none-any.whl (17 kB)\n",
            "Downloading jinja2-3.1.6-py3-none-any.whl (134 kB)\n",
            "Downloading pulp-3.1.1-py3-none-any.whl (16.4 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m16.4/16.4 MB\u001b[0m \u001b[31m81.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading PyYAML-6.0.2-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (759 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m759.5/759.5 kB\u001b[0m \u001b[31m39.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading smart_open-7.1.0-py3-none-any.whl (61 kB)\n",
            "Downloading snakemake_interface_common-1.18.0-py3-none-any.whl (14 kB)\n",
            "Downloading configargparse-1.7.1-py3-none-any.whl (25 kB)\n",
            "Downloading snakemake_interface_executor_plugins-9.3.5-py3-none-any.whl (22 kB)\n",
            "Downloading snakemake_interface_logger_plugins-1.2.3-py3-none-any.whl (6.2 kB)\n",
            "Downloading snakemake_interface_report_plugins-1.1.0-py3-none-any.whl (7.1 kB)\n",
            "Downloading snakemake_interface_storage_plugins-4.2.1-py3-none-any.whl (16 kB)\n",
            "Downloading reretry-0.11.8-py2.py3-none-any.whl (5.6 kB)\n",
            "Downloading throttler-1.2.2-py3-none-any.whl (7.6 kB)\n",
            "Downloading wrapt-1.17.2-cp313-cp313-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (89 kB)\n",
            "Downloading yte-1.7.0-py3-none-any.whl (8.8 kB)\n",
            "Downloading appdirs-1.4.4-py2.py3-none-any.whl (9.6 kB)\n",
            "Downloading docutils-0.21.2-py3-none-any.whl (587 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m587.4/587.4 kB\u001b[0m \u001b[31m27.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading GitPython-3.1.44-py3-none-any.whl (207 kB)\n",
            "Downloading humanfriendly-10.0-py2.py3-none-any.whl (86 kB)\n",
            "Downloading immutables-0.21-cp313-cp313-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (104 kB)\n",
            "Downloading jsonschema-4.24.0-py3-none-any.whl (88 kB)\n",
            "Downloading nbformat-5.10.4-py3-none-any.whl (78 kB)\n",
            "Downloading psutil-7.0.0-cp36-abi3-manylinux_2_12_x86_64.manylinux2010_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (277 kB)\n",
            "Downloading tabulate-0.9.0-py3-none-any.whl (35 kB)\n",
            "Downloading argparse_dataclass-2.0.0-py3-none-any.whl (8.8 kB)\n",
            "Downloading attrs-25.3.0-py3-none-any.whl (63 kB)\n",
            "Downloading fastjsonschema-2.21.1-py3-none-any.whl (23 kB)\n",
            "Downloading gitdb-4.0.12-py3-none-any.whl (62 kB)\n",
            "Downloading jsonschema_specifications-2025.4.1-py3-none-any.whl (18 kB)\n",
            "Downloading jupyter_core-5.8.0-py3-none-any.whl (29 kB)\n",
            "Downloading MarkupSafe-3.0.2-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (23 kB)\n",
            "Downloading plac-1.4.5-py2.py3-none-any.whl (22 kB)\n",
            "Downloading referencing-0.36.2-py3-none-any.whl (26 kB)\n",
            "Downloading rpds_py-0.25.1-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (390 kB)\n",
            "Downloading traitlets-5.14.3-py3-none-any.whl (85 kB)\n",
            "Downloading smmap-5.0.2-py3-none-any.whl (24 kB)\n",
            "Building wheels for collected packages: connection_pool\n",
            "  Building wheel for connection_pool (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for connection_pool: filename=connection_pool-0.0.3-py3-none-any.whl size=4105 sha256=f7822ba19c574a882801e2a2ab0aeffc5f0093254849f91dc8e5a9358dbb3095\n",
            "  Stored in directory: /root/.cache/pip/wheels/eb/50/fe/561652a5d23abeb18e1be0bd933c375c0eec84c4c490707756\n",
            "Successfully built connection_pool\n",
            "Installing collected packages: throttler, plac, fastjsonschema, connection_pool, appdirs, wrapt, traitlets, tabulate, smmap, rpds-py, reretry, pyyaml, pulp, psutil, MarkupSafe, immutables, humanfriendly, dpath, docutils, configargparse, attrs, argparse-dataclass, yte, snakemake-interface-common, smart-open, referencing, jupyter-core, jinja2, gitdb, conda-inject, snakemake-interface-storage-plugins, snakemake-interface-report-plugins, snakemake-interface-logger-plugins, snakemake-interface-executor-plugins, jsonschema-specifications, gitpython, jsonschema, nbformat, snakemake\n",
            "Successfully installed MarkupSafe-3.0.2 appdirs-1.4.4 argparse-dataclass-2.0.0 attrs-25.3.0 conda-inject-1.3.2 configargparse-1.7.1 connection_pool-0.0.3 docutils-0.21.2 dpath-2.2.0 fastjsonschema-2.21.1 gitdb-4.0.12 gitpython-3.1.44 humanfriendly-10.0 immutables-0.21 jinja2-3.1.6 jsonschema-4.24.0 jsonschema-specifications-2025.4.1 jupyter-core-5.8.0 nbformat-5.10.4 plac-1.4.5 psutil-7.0.0 pulp-3.1.1 pyyaml-6.0.2 referencing-0.36.2 reretry-0.11.8 rpds-py-0.25.1 smart-open-7.1.0 smmap-5.0.2 snakemake-9.5.1 snakemake-interface-common-1.18.0 snakemake-interface-executor-plugins-9.3.5 snakemake-interface-logger-plugins-1.2.3 snakemake-interface-report-plugins-1.1.0 snakemake-interface-storage-plugins-4.2.1 tabulate-0.9.0 throttler-1.2.2 traitlets-5.14.3 wrapt-1.17.2 yte-1.7.0\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "#simple esample"
      ],
      "metadata": {
        "id": "5GwMJCUKVdGK"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://ftp.ensemblgenomes.ebi.ac.uk/pub/plants/release-61/fasta/solanum_lycopersicum/dna/Solanum_lycopersicum.SL3.0.dna.toplevel.fa.gz\n",
        "!wget https://ftp.ensemblgenomes.ebi.ac.uk/pub/plants/release-61/gff3/solanum_lycopersicum/Solanum_lycopersicum.SL3.0.61.chr.gff3.gz\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "xbmCoiQTtWJp",
        "outputId": "b7a8471a-ced5-4019-b2a6-70dab42a983d"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2025-05-27 06:39:09--  https://ftp.ensemblgenomes.ebi.ac.uk/pub/plants/release-61/fasta/solanum_lycopersicum/dna/Solanum_lycopersicum.SL3.0.dna.toplevel.fa.gz\n",
            "Resolving ftp.ensemblgenomes.ebi.ac.uk (ftp.ensemblgenomes.ebi.ac.uk)... 193.62.193.161\n",
            "Connecting to ftp.ensemblgenomes.ebi.ac.uk (ftp.ensemblgenomes.ebi.ac.uk)|193.62.193.161|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 227747063 (217M) [application/x-gzip]\n",
            "Saving to: ‘Solanum_lycopersicum.SL3.0.dna.toplevel.fa.gz’\n",
            "\n",
            "Solanum_lycopersicu 100%[===================>] 217.20M  42.2MB/s    in 6.0s    \n",
            "\n",
            "2025-05-27 06:39:16 (36.4 MB/s) - ‘Solanum_lycopersicum.SL3.0.dna.toplevel.fa.gz’ saved [227747063/227747063]\n",
            "\n",
            "--2025-05-27 06:39:16--  https://ftp.ensemblgenomes.ebi.ac.uk/pub/plants/release-61/gff3/solanum_lycopersicum/Solanum_lycopersicum.SL3.0.61.chr.gff3.gz\n",
            "Resolving ftp.ensemblgenomes.ebi.ac.uk (ftp.ensemblgenomes.ebi.ac.uk)... 193.62.193.161\n",
            "Connecting to ftp.ensemblgenomes.ebi.ac.uk (ftp.ensemblgenomes.ebi.ac.uk)|193.62.193.161|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 6448195 (6.1M) [application/x-gzip]\n",
            "Saving to: ‘Solanum_lycopersicum.SL3.0.61.chr.gff3.gz’\n",
            "\n",
            "Solanum_lycopersicu 100%[===================>]   6.15M  8.76MB/s    in 0.7s    \n",
            "\n",
            "2025-05-27 06:39:17 (8.76 MB/s) - ‘Solanum_lycopersicum.SL3.0.61.chr.gff3.gz’ saved [6448195/6448195]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!mkdir resources\n",
        "!gunzip /content/Solanum_lycopersicum.SL3.0.61.chr.gff3.gz\n",
        "!gunzip /content/Solanum_lycopersicum.SL3.0.dna.toplevel.fa.gz"
      ],
      "metadata": {
        "id": "G0_4WiCctwkv"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "%%writefile config.yaml\n",
        "# Reference files\n",
        "genome: \"resources/Solanum_lycopersicum.SL3.0.dna.toplevel.fa\"\n",
        "annotation: \"resources/Solanum_lycopersicum.SL3.0.61.chr.gff3\"\n",
        "\n",
        "# Trimmomatic parameters\n",
        "trimmomatic:\n",
        "  adapters: \"TruSeq3-SE.fa\"\n",
        "  leading: 20\n",
        "  trailing: 20\n",
        "  sliding_window: \"4:20\"\n",
        "  minlen: 36\n",
        "\n",
        "# HTSeq parameters\n",
        "htseq:\n",
        "  stranded: \"no\"\n",
        "  idattr: \"ID\"\n",
        "\n",
        "#hist2 installtion\n",
        "name: snakemake\n",
        "channels:\n",
        "  - conda-forge\n",
        "  - bioconda\n",
        "dependencies:\n",
        "  - snakemake = 9.5.1\n",
        "  - hisat2 = 2.2.0\n",
        "  - samtools = 1.21\n",
        "\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "lQhtOGNmsuEx",
        "outputId": "0bf70d55-31a0-4da7-f717-ee07d777ffd5"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Overwriting config.yaml\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "%%writefile Snakefile\n",
        "# Snakefile for Tomato RNA-seq Analysis Pipeline\n",
        "import os\n",
        "\n",
        "configfile: \"config.yaml\"\n",
        "\n",
        "# Define SRA accessions\n",
        "SRR_IDS = [f\"SRR987959{i}\" for i in range(4, 10)] + [f\"SRR98796{i}\" for i in range(0, 6)]\n",
        "\n",
        "# Reference files\n",
        "GENOME = config[\"genome\"]\n",
        "GTF = config[\"annotation\"]\n",
        "\n",
        "rule all:\n",
        "    input:\n",
        "        expand(\"results/counts/{srr}.counts.txt\", srr=SRR_IDS),\n",
        "        \"multiqc_report.html\"\n",
        "\n",
        "# SRA Download\n",
        "rule download_sra:\n",
        "    output:\n",
        "        \"data/raw/{srr}.fastq\"\n",
        "    params:\n",
        "        prefetch_dir=temp(\"data/raw/.prefetch/{srr}\")\n",
        "    log:\n",
        "        \"logs/sra/{srr}.log\"\n",
        "    wrapper:\n",
        "        \"v1.2.0/bio/sra-tools/fasterq-dump\"\n",
        "\n",
        "# FastQC Raw\n",
        "rule fastqc_raw:\n",
        "    input:\n",
        "        \"data/raw/{srr}.fastq\"\n",
        "    output:\n",
        "        html=\"results/fastqc/raw/{srr}_fastqc.html\",\n",
        "        zip=\"results/fastqc/raw/{srr}_fastqc.zip\"\n",
        "    log:\n",
        "        \"logs/fastqc/raw/{srr}.log\"\n",
        "    threads: 2\n",
        "    wrapper:\n",
        "        \"v1.2.0/bio/fastqc\"\n",
        "\n",
        "# Trimmomatic\n",
        "rule trim:\n",
        "    input:\n",
        "        \"data/raw/{srr}.fastq\"\n",
        "    output:\n",
        "        trimmed=\"data/trimmed/{srr}.trimmed.fastq\",\n",
        "        unpaired=\"data/trimmed/{srr}.unpaired.fastq\"\n",
        "    params:\n",
        "        adapters=config[\"trimmomatic\"][\"adapters\"],\n",
        "        leading=20,\n",
        "        trailing=20,\n",
        "        sliding_window=\"4:20\",\n",
        "        minlen=36\n",
        "    log:\n",
        "        \"logs/trimmomatic/{srr}.log\"\n",
        "    threads: 4\n",
        "    wrapper:\n",
        "        \"v1.2.0/bio/trimmomatic/se\"\n",
        "\n",
        "# FastQC Trimmed\n",
        "rule fastqc_trimmed:\n",
        "    input:\n",
        "        \"data/trimmed/{srr}.trimmed.fastq\"\n",
        "    output:\n",
        "        html=\"results/fastqc/trimmed/{srr}_trimmed_fastqc.html\",\n",
        "        zip=\"results/fastqc/trimmed/{srr}_trimmed_fastqc.zip\"\n",
        "    log:\n",
        "        \"logs/fastqc/trimmed/{srr}.log\"\n",
        "    threads: 2\n",
        "    wrapper:\n",
        "        \"v1.2.0/bio/fastqc\"\n",
        "\n",
        "# HISAT2 Index\n",
        "rule hisat2_index:\n",
        "    input:\n",
        "        \"resources/Solanum_lycopersicum.SL3.0.dna.toplevel.fa\"\n",
        "    output:\n",
        "        index=directory(\"resources/hisat2_index\")\n",
        "    params:\n",
        "        prefix=\"resources/hisat2_index/tomato\"\n",
        "    log:\n",
        "        \"logs/hisat2/index.log\"\n",
        "    threads: 8\n",
        "    shell:\n",
        "        \"\"\"\n",
        "        mkdir -p logs/hisat2\n",
        "        hisat2-build -p {threads} {input} reference_index {params.prefix}  > {log} 2>&1\n",
        "        \"\"\"\n",
        "\n",
        "# HISAT2 Alignment\n",
        "rule hisat2_align:\n",
        "    input:\n",
        "        fastq=\"data/trimmed/{srr}.trimmed.fastq\",\n",
        "        index=directory(\"resources/hisat2_index\")\n",
        "    output:\n",
        "        \"results/aligned/{srr}.sam\"\n",
        "    params:\n",
        "        index_prefix=\"resources/hisat2_index/tomato\"\n",
        "    log:\n",
        "        \"logs/hisat2/{srr}.log\"\n",
        "    threads: 8\n",
        "    shell:\n",
        "        \"hisat2 -x {params.index_prefix} -U {input.fastq} -S {output} \"\n",
        "        \"--threads {threads} --dta --summary-file {log}\"\n",
        "\n",
        "# SAM to BAM\n",
        "rule samtools_sort:\n",
        "    input:\n",
        "        \"results/aligned/{srr}.sam\"\n",
        "    output:\n",
        "        \"results/aligned/{srr}.sorted.bam\"\n",
        "    log:\n",
        "        \"logs/samtools/{srr}.sort.log\"\n",
        "    threads: 4\n",
        "    shell:\n",
        "        \"samtools sort -@ {threads} -o {output} {input} > {log} 2>&1\"\n",
        "\n",
        "\n",
        "# HTSeq Count\n",
        "rule htseq_count:\n",
        "    input:\n",
        "        bam=\"results/aligned/{srr}.sorted.bam\",\n",
        "        gff=GTF\n",
        "    output:\n",
        "        \"results/counts/{srr}.counts.txt\"\n",
        "    log:\n",
        "        \"logs/htseq/{srr}.log\"\n",
        "    threads: 1\n",
        "    shell:\n",
        "        \"\"\"\n",
        "        htseq-count -f bam -r pos -s no -t exon -i ID {input.bam} {input.gff} > {output} 2> {log}\n",
        "        \"\"\"\n",
        "\n",
        "\n",
        "# MultiQC Report\n",
        "rule multiqc:\n",
        "    input:\n",
        "        fastqc_raw=expand(\"results/fastqc/raw/{srr}_fastqc.zip\", srr=SRR_IDS),\n",
        "        fastqc_trimmed=expand(\"results/fastqc/trimmed/{srr}_trimmed_fastqc.zip\", srr=SRR_IDS),\n",
        "        trimmomatic_logs=expand(\"logs/trimmomatic/{srr}.log\", srr=SRR_IDS),\n",
        "        bam_files=expand(\"results/aligned/{srr}.sorted.bam\", srr=SRR_IDS)\n",
        "    output:\n",
        "        \"multiqc_report.html\"\n",
        "    shell:\n",
        "        \"\"\"\n",
        "        multiqc results/fastqc/raw results/fastqc/trimmed logs/trimmomatic results/aligned -o .\n",
        "        \"\"\"\n",
        "\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "emllvhJXVjDT",
        "outputId": "89747682-9400-4972-a846-8f1433025140"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Overwriting Snakefile\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!mv Solanum_lycopersicum.SL3.0.dna.toplevel.fa Solanum_lycopersicum.SL3.0.61.chr.gff3 resources/"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "K6dk3hNcp0cW",
        "outputId": "f458ca25-1831-40d9-a65f-99799583094b"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "mv: cannot stat 'Solanum_lycopersicum.SL3.0.dna.toplevel.fa': No such file or directory\n",
            "mv: cannot stat 'Solanum_lycopersicum.SL3.0.61.chr.gff3': No such file or directory\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!snakemake --cores 4 --dry-run"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "UbRDw0XGWY8l",
        "outputId": "d83aa110-dc57-490e-8ddd-16c7ea7bd512",
        "collapsed": true
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "The flag 'directory' used in rule hisat2_align is only valid for outputs, not inputs.\n",
            "\u001b[33mhost: bb74ec2f9ae5\u001b[0m\n",
            "\u001b[33mBuilding DAG of jobs...\u001b[0m\n",
            "\u001b[33mJob stats:\n",
            "job               count\n",
            "--------------  -------\n",
            "all                   1\n",
            "download_sra         12\n",
            "fastqc_raw           12\n",
            "fastqc_trimmed       12\n",
            "hisat2_align         12\n",
            "hisat2_index          1\n",
            "htseq_count          12\n",
            "multiqc               1\n",
            "samtools_sort        12\n",
            "trim                 12\n",
            "total                87\n",
            "\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule download_sra:\n",
            "    output: data/raw/SRR987962.fastq\n",
            "    log: logs/sra/SRR987962.log\n",
            "    jobid: 46\n",
            "    reason: Missing output files: data/raw/SRR987962.fastq\n",
            "    wildcards: srr=SRR987962\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule download_sra:\n",
            "    output: data/raw/SRR987964.fastq\n",
            "    log: logs/sra/SRR987964.log\n",
            "    jobid: 56\n",
            "    reason: Missing output files: data/raw/SRR987964.fastq\n",
            "    wildcards: srr=SRR987964\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule download_sra:\n",
            "    output: data/raw/SRR9879598.fastq\n",
            "    log: logs/sra/SRR9879598.log\n",
            "    jobid: 26\n",
            "    reason: Missing output files: data/raw/SRR9879598.fastq\n",
            "    wildcards: srr=SRR9879598\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule download_sra:\n",
            "    output: data/raw/SRR9879597.fastq\n",
            "    log: logs/sra/SRR9879597.log\n",
            "    jobid: 21\n",
            "    reason: Missing output files: data/raw/SRR9879597.fastq\n",
            "    wildcards: srr=SRR9879597\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule download_sra:\n",
            "    output: data/raw/SRR9879594.fastq\n",
            "    log: logs/sra/SRR9879594.log\n",
            "    jobid: 5\n",
            "    reason: Missing output files: data/raw/SRR9879594.fastq\n",
            "    wildcards: srr=SRR9879594\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule download_sra:\n",
            "    output: data/raw/SRR987965.fastq\n",
            "    log: logs/sra/SRR987965.log\n",
            "    jobid: 61\n",
            "    reason: Missing output files: data/raw/SRR987965.fastq\n",
            "    wildcards: srr=SRR987965\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule download_sra:\n",
            "    output: data/raw/SRR9879595.fastq\n",
            "    log: logs/sra/SRR9879595.log\n",
            "    jobid: 11\n",
            "    reason: Missing output files: data/raw/SRR9879595.fastq\n",
            "    wildcards: srr=SRR9879595\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule download_sra:\n",
            "    output: data/raw/SRR987960.fastq\n",
            "    log: logs/sra/SRR987960.log\n",
            "    jobid: 36\n",
            "    reason: Missing output files: data/raw/SRR987960.fastq\n",
            "    wildcards: srr=SRR987960\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule download_sra:\n",
            "    output: data/raw/SRR9879599.fastq\n",
            "    log: logs/sra/SRR9879599.log\n",
            "    jobid: 31\n",
            "    reason: Missing output files: data/raw/SRR9879599.fastq\n",
            "    wildcards: srr=SRR9879599\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule download_sra:\n",
            "    output: data/raw/SRR9879596.fastq\n",
            "    log: logs/sra/SRR9879596.log\n",
            "    jobid: 16\n",
            "    reason: Missing output files: data/raw/SRR9879596.fastq\n",
            "    wildcards: srr=SRR9879596\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule download_sra:\n",
            "    output: data/raw/SRR987961.fastq\n",
            "    log: logs/sra/SRR987961.log\n",
            "    jobid: 41\n",
            "    reason: Missing output files: data/raw/SRR987961.fastq\n",
            "    wildcards: srr=SRR987961\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule hisat2_index:\n",
            "    input: resources/Solanum_lycopersicum.SL3.0.dna.toplevel.fa\n",
            "    output: resources/hisat2_index\n",
            "    log: logs/hisat2/index.log\n",
            "    jobid: 6\n",
            "    reason: Missing output files: resources/hisat2_index\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule download_sra:\n",
            "    output: data/raw/SRR987963.fastq\n",
            "    log: logs/sra/SRR987963.log\n",
            "    jobid: 51\n",
            "    reason: Missing output files: data/raw/SRR987963.fastq\n",
            "    wildcards: srr=SRR987963\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule fastqc_raw:\n",
            "    input: data/raw/SRR987964.fastq\n",
            "    output: results/fastqc/raw/SRR987964_fastqc.html, results/fastqc/raw/SRR987964_fastqc.zip\n",
            "    log: logs/fastqc/raw/SRR987964.log\n",
            "    jobid: 73\n",
            "    reason: Missing output files: results/fastqc/raw/SRR987964_fastqc.zip; Input files updated by another job: data/raw/SRR987964.fastq\n",
            "    wildcards: srr=SRR987964\n",
            "    threads: 2\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule trim:\n",
            "    input: data/raw/SRR987965.fastq\n",
            "    output: data/trimmed/SRR987965.trimmed.fastq, data/trimmed/SRR987965.unpaired.fastq\n",
            "    log: logs/trimmomatic/SRR987965.log\n",
            "    jobid: 60\n",
            "    reason: Missing output files: logs/trimmomatic/SRR987965.log, data/trimmed/SRR987965.trimmed.fastq; Input files updated by another job: data/raw/SRR987965.fastq\n",
            "    wildcards: srr=SRR987965\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule fastqc_raw:\n",
            "    input: data/raw/SRR987962.fastq\n",
            "    output: results/fastqc/raw/SRR987962_fastqc.html, results/fastqc/raw/SRR987962_fastqc.zip\n",
            "    log: logs/fastqc/raw/SRR987962.log\n",
            "    jobid: 71\n",
            "    reason: Missing output files: results/fastqc/raw/SRR987962_fastqc.zip; Input files updated by another job: data/raw/SRR987962.fastq\n",
            "    wildcards: srr=SRR987962\n",
            "    threads: 2\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule fastqc_raw:\n",
            "    input: data/raw/SRR9879595.fastq\n",
            "    output: results/fastqc/raw/SRR9879595_fastqc.html, results/fastqc/raw/SRR9879595_fastqc.zip\n",
            "    log: logs/fastqc/raw/SRR9879595.log\n",
            "    jobid: 64\n",
            "    reason: Missing output files: results/fastqc/raw/SRR9879595_fastqc.zip; Input files updated by another job: data/raw/SRR9879595.fastq\n",
            "    wildcards: srr=SRR9879595\n",
            "    threads: 2\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule trim:\n",
            "    input: data/raw/SRR9879598.fastq\n",
            "    output: data/trimmed/SRR9879598.trimmed.fastq, data/trimmed/SRR9879598.unpaired.fastq\n",
            "    log: logs/trimmomatic/SRR9879598.log\n",
            "    jobid: 25\n",
            "    reason: Missing output files: logs/trimmomatic/SRR9879598.log, data/trimmed/SRR9879598.trimmed.fastq; Input files updated by another job: data/raw/SRR9879598.fastq\n",
            "    wildcards: srr=SRR9879598\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule trim:\n",
            "    input: data/raw/SRR987964.fastq\n",
            "    output: data/trimmed/SRR987964.trimmed.fastq, data/trimmed/SRR987964.unpaired.fastq\n",
            "    log: logs/trimmomatic/SRR987964.log\n",
            "    jobid: 55\n",
            "    reason: Missing output files: logs/trimmomatic/SRR987964.log, data/trimmed/SRR987964.trimmed.fastq; Input files updated by another job: data/raw/SRR987964.fastq\n",
            "    wildcards: srr=SRR987964\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule fastqc_raw:\n",
            "    input: data/raw/SRR987960.fastq\n",
            "    output: results/fastqc/raw/SRR987960_fastqc.html, results/fastqc/raw/SRR987960_fastqc.zip\n",
            "    log: logs/fastqc/raw/SRR987960.log\n",
            "    jobid: 69\n",
            "    reason: Missing output files: results/fastqc/raw/SRR987960_fastqc.zip; Input files updated by another job: data/raw/SRR987960.fastq\n",
            "    wildcards: srr=SRR987960\n",
            "    threads: 2\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule trim:\n",
            "    input: data/raw/SRR987961.fastq\n",
            "    output: data/trimmed/SRR987961.trimmed.fastq, data/trimmed/SRR987961.unpaired.fastq\n",
            "    log: logs/trimmomatic/SRR987961.log\n",
            "    jobid: 40\n",
            "    reason: Missing output files: logs/trimmomatic/SRR987961.log, data/trimmed/SRR987961.trimmed.fastq; Input files updated by another job: data/raw/SRR987961.fastq\n",
            "    wildcards: srr=SRR987961\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule fastqc_raw:\n",
            "    input: data/raw/SRR9879598.fastq\n",
            "    output: results/fastqc/raw/SRR9879598_fastqc.html, results/fastqc/raw/SRR9879598_fastqc.zip\n",
            "    log: logs/fastqc/raw/SRR9879598.log\n",
            "    jobid: 67\n",
            "    reason: Missing output files: results/fastqc/raw/SRR9879598_fastqc.zip; Input files updated by another job: data/raw/SRR9879598.fastq\n",
            "    wildcards: srr=SRR9879598\n",
            "    threads: 2\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule trim:\n",
            "    input: data/raw/SRR9879599.fastq\n",
            "    output: data/trimmed/SRR9879599.trimmed.fastq, data/trimmed/SRR9879599.unpaired.fastq\n",
            "    log: logs/trimmomatic/SRR9879599.log\n",
            "    jobid: 30\n",
            "    reason: Missing output files: data/trimmed/SRR9879599.trimmed.fastq, logs/trimmomatic/SRR9879599.log; Input files updated by another job: data/raw/SRR9879599.fastq\n",
            "    wildcards: srr=SRR9879599\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule trim:\n",
            "    input: data/raw/SRR9879595.fastq\n",
            "    output: data/trimmed/SRR9879595.trimmed.fastq, data/trimmed/SRR9879595.unpaired.fastq\n",
            "    log: logs/trimmomatic/SRR9879595.log\n",
            "    jobid: 10\n",
            "    reason: Missing output files: logs/trimmomatic/SRR9879595.log, data/trimmed/SRR9879595.trimmed.fastq; Input files updated by another job: data/raw/SRR9879595.fastq\n",
            "    wildcards: srr=SRR9879595\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule fastqc_raw:\n",
            "    input: data/raw/SRR987965.fastq\n",
            "    output: results/fastqc/raw/SRR987965_fastqc.html, results/fastqc/raw/SRR987965_fastqc.zip\n",
            "    log: logs/fastqc/raw/SRR987965.log\n",
            "    jobid: 74\n",
            "    reason: Missing output files: results/fastqc/raw/SRR987965_fastqc.zip; Input files updated by another job: data/raw/SRR987965.fastq\n",
            "    wildcards: srr=SRR987965\n",
            "    threads: 2\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule trim:\n",
            "    input: data/raw/SRR987962.fastq\n",
            "    output: data/trimmed/SRR987962.trimmed.fastq, data/trimmed/SRR987962.unpaired.fastq\n",
            "    log: logs/trimmomatic/SRR987962.log\n",
            "    jobid: 45\n",
            "    reason: Missing output files: logs/trimmomatic/SRR987962.log, data/trimmed/SRR987962.trimmed.fastq; Input files updated by another job: data/raw/SRR987962.fastq\n",
            "    wildcards: srr=SRR987962\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule fastqc_raw:\n",
            "    input: data/raw/SRR987963.fastq\n",
            "    output: results/fastqc/raw/SRR987963_fastqc.html, results/fastqc/raw/SRR987963_fastqc.zip\n",
            "    log: logs/fastqc/raw/SRR987963.log\n",
            "    jobid: 72\n",
            "    reason: Missing output files: results/fastqc/raw/SRR987963_fastqc.zip; Input files updated by another job: data/raw/SRR987963.fastq\n",
            "    wildcards: srr=SRR987963\n",
            "    threads: 2\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule fastqc_raw:\n",
            "    input: data/raw/SRR9879596.fastq\n",
            "    output: results/fastqc/raw/SRR9879596_fastqc.html, results/fastqc/raw/SRR9879596_fastqc.zip\n",
            "    log: logs/fastqc/raw/SRR9879596.log\n",
            "    jobid: 65\n",
            "    reason: Missing output files: results/fastqc/raw/SRR9879596_fastqc.zip; Input files updated by another job: data/raw/SRR9879596.fastq\n",
            "    wildcards: srr=SRR9879596\n",
            "    threads: 2\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule trim:\n",
            "    input: data/raw/SRR9879594.fastq\n",
            "    output: data/trimmed/SRR9879594.trimmed.fastq, data/trimmed/SRR9879594.unpaired.fastq\n",
            "    log: logs/trimmomatic/SRR9879594.log\n",
            "    jobid: 4\n",
            "    reason: Missing output files: logs/trimmomatic/SRR9879594.log, data/trimmed/SRR9879594.trimmed.fastq; Input files updated by another job: data/raw/SRR9879594.fastq\n",
            "    wildcards: srr=SRR9879594\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule trim:\n",
            "    input: data/raw/SRR9879596.fastq\n",
            "    output: data/trimmed/SRR9879596.trimmed.fastq, data/trimmed/SRR9879596.unpaired.fastq\n",
            "    log: logs/trimmomatic/SRR9879596.log\n",
            "    jobid: 15\n",
            "    reason: Missing output files: logs/trimmomatic/SRR9879596.log, data/trimmed/SRR9879596.trimmed.fastq; Input files updated by another job: data/raw/SRR9879596.fastq\n",
            "    wildcards: srr=SRR9879596\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule fastqc_raw:\n",
            "    input: data/raw/SRR987961.fastq\n",
            "    output: results/fastqc/raw/SRR987961_fastqc.html, results/fastqc/raw/SRR987961_fastqc.zip\n",
            "    log: logs/fastqc/raw/SRR987961.log\n",
            "    jobid: 70\n",
            "    reason: Missing output files: results/fastqc/raw/SRR987961_fastqc.zip; Input files updated by another job: data/raw/SRR987961.fastq\n",
            "    wildcards: srr=SRR987961\n",
            "    threads: 2\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule fastqc_raw:\n",
            "    input: data/raw/SRR9879594.fastq\n",
            "    output: results/fastqc/raw/SRR9879594_fastqc.html, results/fastqc/raw/SRR9879594_fastqc.zip\n",
            "    log: logs/fastqc/raw/SRR9879594.log\n",
            "    jobid: 63\n",
            "    reason: Missing output files: results/fastqc/raw/SRR9879594_fastqc.zip; Input files updated by another job: data/raw/SRR9879594.fastq\n",
            "    wildcards: srr=SRR9879594\n",
            "    threads: 2\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule trim:\n",
            "    input: data/raw/SRR9879597.fastq\n",
            "    output: data/trimmed/SRR9879597.trimmed.fastq, data/trimmed/SRR9879597.unpaired.fastq\n",
            "    log: logs/trimmomatic/SRR9879597.log\n",
            "    jobid: 20\n",
            "    reason: Missing output files: logs/trimmomatic/SRR9879597.log, data/trimmed/SRR9879597.trimmed.fastq; Input files updated by another job: data/raw/SRR9879597.fastq\n",
            "    wildcards: srr=SRR9879597\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule trim:\n",
            "    input: data/raw/SRR987963.fastq\n",
            "    output: data/trimmed/SRR987963.trimmed.fastq, data/trimmed/SRR987963.unpaired.fastq\n",
            "    log: logs/trimmomatic/SRR987963.log\n",
            "    jobid: 50\n",
            "    reason: Missing output files: data/trimmed/SRR987963.trimmed.fastq, logs/trimmomatic/SRR987963.log; Input files updated by another job: data/raw/SRR987963.fastq\n",
            "    wildcards: srr=SRR987963\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule fastqc_raw:\n",
            "    input: data/raw/SRR9879599.fastq\n",
            "    output: results/fastqc/raw/SRR9879599_fastqc.html, results/fastqc/raw/SRR9879599_fastqc.zip\n",
            "    log: logs/fastqc/raw/SRR9879599.log\n",
            "    jobid: 68\n",
            "    reason: Missing output files: results/fastqc/raw/SRR9879599_fastqc.zip; Input files updated by another job: data/raw/SRR9879599.fastq\n",
            "    wildcards: srr=SRR9879599\n",
            "    threads: 2\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule fastqc_raw:\n",
            "    input: data/raw/SRR9879597.fastq\n",
            "    output: results/fastqc/raw/SRR9879597_fastqc.html, results/fastqc/raw/SRR9879597_fastqc.zip\n",
            "    log: logs/fastqc/raw/SRR9879597.log\n",
            "    jobid: 66\n",
            "    reason: Missing output files: results/fastqc/raw/SRR9879597_fastqc.zip; Input files updated by another job: data/raw/SRR9879597.fastq\n",
            "    wildcards: srr=SRR9879597\n",
            "    threads: 2\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule trim:\n",
            "    input: data/raw/SRR987960.fastq\n",
            "    output: data/trimmed/SRR987960.trimmed.fastq, data/trimmed/SRR987960.unpaired.fastq\n",
            "    log: logs/trimmomatic/SRR987960.log\n",
            "    jobid: 35\n",
            "    reason: Missing output files: data/trimmed/SRR987960.trimmed.fastq, logs/trimmomatic/SRR987960.log; Input files updated by another job: data/raw/SRR987960.fastq\n",
            "    wildcards: srr=SRR987960\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule fastqc_trimmed:\n",
            "    input: data/trimmed/SRR9879599.trimmed.fastq\n",
            "    output: results/fastqc/trimmed/SRR9879599_trimmed_fastqc.html, results/fastqc/trimmed/SRR9879599_trimmed_fastqc.zip\n",
            "    log: logs/fastqc/trimmed/SRR9879599.log\n",
            "    jobid: 80\n",
            "    reason: Missing output files: results/fastqc/trimmed/SRR9879599_trimmed_fastqc.zip; Input files updated by another job: data/trimmed/SRR9879599.trimmed.fastq\n",
            "    wildcards: srr=SRR9879599\n",
            "    threads: 2\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule hisat2_align:\n",
            "    input: data/trimmed/SRR9879595.trimmed.fastq, resources/hisat2_index\n",
            "    output: results/aligned/SRR9879595.sam\n",
            "    log: logs/hisat2/SRR9879595.log\n",
            "    jobid: 9\n",
            "    reason: Missing output files: results/aligned/SRR9879595.sam; Input files updated by another job: resources/hisat2_index, data/trimmed/SRR9879595.trimmed.fastq\n",
            "    wildcards: srr=SRR9879595\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule fastqc_trimmed:\n",
            "    input: data/trimmed/SRR9879597.trimmed.fastq\n",
            "    output: results/fastqc/trimmed/SRR9879597_trimmed_fastqc.html, results/fastqc/trimmed/SRR9879597_trimmed_fastqc.zip\n",
            "    log: logs/fastqc/trimmed/SRR9879597.log\n",
            "    jobid: 78\n",
            "    reason: Missing output files: results/fastqc/trimmed/SRR9879597_trimmed_fastqc.zip; Input files updated by another job: data/trimmed/SRR9879597.trimmed.fastq\n",
            "    wildcards: srr=SRR9879597\n",
            "    threads: 2\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule hisat2_align:\n",
            "    input: data/trimmed/SRR987962.trimmed.fastq, resources/hisat2_index\n",
            "    output: results/aligned/SRR987962.sam\n",
            "    log: logs/hisat2/SRR987962.log\n",
            "    jobid: 44\n",
            "    reason: Missing output files: results/aligned/SRR987962.sam; Input files updated by another job: resources/hisat2_index, data/trimmed/SRR987962.trimmed.fastq\n",
            "    wildcards: srr=SRR987962\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule hisat2_align:\n",
            "    input: data/trimmed/SRR9879594.trimmed.fastq, resources/hisat2_index\n",
            "    output: results/aligned/SRR9879594.sam\n",
            "    log: logs/hisat2/SRR9879594.log\n",
            "    jobid: 3\n",
            "    reason: Missing output files: results/aligned/SRR9879594.sam; Input files updated by another job: resources/hisat2_index, data/trimmed/SRR9879594.trimmed.fastq\n",
            "    wildcards: srr=SRR9879594\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule fastqc_trimmed:\n",
            "    input: data/trimmed/SRR987964.trimmed.fastq\n",
            "    output: results/fastqc/trimmed/SRR987964_trimmed_fastqc.html, results/fastqc/trimmed/SRR987964_trimmed_fastqc.zip\n",
            "    log: logs/fastqc/trimmed/SRR987964.log\n",
            "    jobid: 85\n",
            "    reason: Missing output files: results/fastqc/trimmed/SRR987964_trimmed_fastqc.zip; Input files updated by another job: data/trimmed/SRR987964.trimmed.fastq\n",
            "    wildcards: srr=SRR987964\n",
            "    threads: 2\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule hisat2_align:\n",
            "    input: data/trimmed/SRR9879596.trimmed.fastq, resources/hisat2_index\n",
            "    output: results/aligned/SRR9879596.sam\n",
            "    log: logs/hisat2/SRR9879596.log\n",
            "    jobid: 14\n",
            "    reason: Missing output files: results/aligned/SRR9879596.sam; Input files updated by another job: resources/hisat2_index, data/trimmed/SRR9879596.trimmed.fastq\n",
            "    wildcards: srr=SRR9879596\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule fastqc_trimmed:\n",
            "    input: data/trimmed/SRR987962.trimmed.fastq\n",
            "    output: results/fastqc/trimmed/SRR987962_trimmed_fastqc.html, results/fastqc/trimmed/SRR987962_trimmed_fastqc.zip\n",
            "    log: logs/fastqc/trimmed/SRR987962.log\n",
            "    jobid: 83\n",
            "    reason: Missing output files: results/fastqc/trimmed/SRR987962_trimmed_fastqc.zip; Input files updated by another job: data/trimmed/SRR987962.trimmed.fastq\n",
            "    wildcards: srr=SRR987962\n",
            "    threads: 2\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule fastqc_trimmed:\n",
            "    input: data/trimmed/SRR9879595.trimmed.fastq\n",
            "    output: results/fastqc/trimmed/SRR9879595_trimmed_fastqc.html, results/fastqc/trimmed/SRR9879595_trimmed_fastqc.zip\n",
            "    log: logs/fastqc/trimmed/SRR9879595.log\n",
            "    jobid: 76\n",
            "    reason: Missing output files: results/fastqc/trimmed/SRR9879595_trimmed_fastqc.zip; Input files updated by another job: data/trimmed/SRR9879595.trimmed.fastq\n",
            "    wildcards: srr=SRR9879595\n",
            "    threads: 2\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule hisat2_align:\n",
            "    input: data/trimmed/SRR9879597.trimmed.fastq, resources/hisat2_index\n",
            "    output: results/aligned/SRR9879597.sam\n",
            "    log: logs/hisat2/SRR9879597.log\n",
            "    jobid: 19\n",
            "    reason: Missing output files: results/aligned/SRR9879597.sam; Input files updated by another job: resources/hisat2_index, data/trimmed/SRR9879597.trimmed.fastq\n",
            "    wildcards: srr=SRR9879597\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule hisat2_align:\n",
            "    input: data/trimmed/SRR987963.trimmed.fastq, resources/hisat2_index\n",
            "    output: results/aligned/SRR987963.sam\n",
            "    log: logs/hisat2/SRR987963.log\n",
            "    jobid: 49\n",
            "    reason: Missing output files: results/aligned/SRR987963.sam; Input files updated by another job: resources/hisat2_index, data/trimmed/SRR987963.trimmed.fastq\n",
            "    wildcards: srr=SRR987963\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule fastqc_trimmed:\n",
            "    input: data/trimmed/SRR987960.trimmed.fastq\n",
            "    output: results/fastqc/trimmed/SRR987960_trimmed_fastqc.html, results/fastqc/trimmed/SRR987960_trimmed_fastqc.zip\n",
            "    log: logs/fastqc/trimmed/SRR987960.log\n",
            "    jobid: 81\n",
            "    reason: Missing output files: results/fastqc/trimmed/SRR987960_trimmed_fastqc.zip; Input files updated by another job: data/trimmed/SRR987960.trimmed.fastq\n",
            "    wildcards: srr=SRR987960\n",
            "    threads: 2\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule hisat2_align:\n",
            "    input: data/trimmed/SRR987960.trimmed.fastq, resources/hisat2_index\n",
            "    output: results/aligned/SRR987960.sam\n",
            "    log: logs/hisat2/SRR987960.log\n",
            "    jobid: 34\n",
            "    reason: Missing output files: results/aligned/SRR987960.sam; Input files updated by another job: data/trimmed/SRR987960.trimmed.fastq, resources/hisat2_index\n",
            "    wildcards: srr=SRR987960\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule fastqc_trimmed:\n",
            "    input: data/trimmed/SRR987965.trimmed.fastq\n",
            "    output: results/fastqc/trimmed/SRR987965_trimmed_fastqc.html, results/fastqc/trimmed/SRR987965_trimmed_fastqc.zip\n",
            "    log: logs/fastqc/trimmed/SRR987965.log\n",
            "    jobid: 86\n",
            "    reason: Missing output files: results/fastqc/trimmed/SRR987965_trimmed_fastqc.zip; Input files updated by another job: data/trimmed/SRR987965.trimmed.fastq\n",
            "    wildcards: srr=SRR987965\n",
            "    threads: 2\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule fastqc_trimmed:\n",
            "    input: data/trimmed/SRR9879598.trimmed.fastq\n",
            "    output: results/fastqc/trimmed/SRR9879598_trimmed_fastqc.html, results/fastqc/trimmed/SRR9879598_trimmed_fastqc.zip\n",
            "    log: logs/fastqc/trimmed/SRR9879598.log\n",
            "    jobid: 79\n",
            "    reason: Missing output files: results/fastqc/trimmed/SRR9879598_trimmed_fastqc.zip; Input files updated by another job: data/trimmed/SRR9879598.trimmed.fastq\n",
            "    wildcards: srr=SRR9879598\n",
            "    threads: 2\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule fastqc_trimmed:\n",
            "    input: data/trimmed/SRR987963.trimmed.fastq\n",
            "    output: results/fastqc/trimmed/SRR987963_trimmed_fastqc.html, results/fastqc/trimmed/SRR987963_trimmed_fastqc.zip\n",
            "    log: logs/fastqc/trimmed/SRR987963.log\n",
            "    jobid: 84\n",
            "    reason: Missing output files: results/fastqc/trimmed/SRR987963_trimmed_fastqc.zip; Input files updated by another job: data/trimmed/SRR987963.trimmed.fastq\n",
            "    wildcards: srr=SRR987963\n",
            "    threads: 2\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule hisat2_align:\n",
            "    input: data/trimmed/SRR987965.trimmed.fastq, resources/hisat2_index\n",
            "    output: results/aligned/SRR987965.sam\n",
            "    log: logs/hisat2/SRR987965.log\n",
            "    jobid: 59\n",
            "    reason: Missing output files: results/aligned/SRR987965.sam; Input files updated by another job: data/trimmed/SRR987965.trimmed.fastq, resources/hisat2_index\n",
            "    wildcards: srr=SRR987965\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule fastqc_trimmed:\n",
            "    input: data/trimmed/SRR9879596.trimmed.fastq\n",
            "    output: results/fastqc/trimmed/SRR9879596_trimmed_fastqc.html, results/fastqc/trimmed/SRR9879596_trimmed_fastqc.zip\n",
            "    log: logs/fastqc/trimmed/SRR9879596.log\n",
            "    jobid: 77\n",
            "    reason: Missing output files: results/fastqc/trimmed/SRR9879596_trimmed_fastqc.zip; Input files updated by another job: data/trimmed/SRR9879596.trimmed.fastq\n",
            "    wildcards: srr=SRR9879596\n",
            "    threads: 2\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule hisat2_align:\n",
            "    input: data/trimmed/SRR9879598.trimmed.fastq, resources/hisat2_index\n",
            "    output: results/aligned/SRR9879598.sam\n",
            "    log: logs/hisat2/SRR9879598.log\n",
            "    jobid: 24\n",
            "    reason: Missing output files: results/aligned/SRR9879598.sam; Input files updated by another job: resources/hisat2_index, data/trimmed/SRR9879598.trimmed.fastq\n",
            "    wildcards: srr=SRR9879598\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule hisat2_align:\n",
            "    input: data/trimmed/SRR987964.trimmed.fastq, resources/hisat2_index\n",
            "    output: results/aligned/SRR987964.sam\n",
            "    log: logs/hisat2/SRR987964.log\n",
            "    jobid: 54\n",
            "    reason: Missing output files: results/aligned/SRR987964.sam; Input files updated by another job: resources/hisat2_index, data/trimmed/SRR987964.trimmed.fastq\n",
            "    wildcards: srr=SRR987964\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule fastqc_trimmed:\n",
            "    input: data/trimmed/SRR987961.trimmed.fastq\n",
            "    output: results/fastqc/trimmed/SRR987961_trimmed_fastqc.html, results/fastqc/trimmed/SRR987961_trimmed_fastqc.zip\n",
            "    log: logs/fastqc/trimmed/SRR987961.log\n",
            "    jobid: 82\n",
            "    reason: Missing output files: results/fastqc/trimmed/SRR987961_trimmed_fastqc.zip; Input files updated by another job: data/trimmed/SRR987961.trimmed.fastq\n",
            "    wildcards: srr=SRR987961\n",
            "    threads: 2\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule fastqc_trimmed:\n",
            "    input: data/trimmed/SRR9879594.trimmed.fastq\n",
            "    output: results/fastqc/trimmed/SRR9879594_trimmed_fastqc.html, results/fastqc/trimmed/SRR9879594_trimmed_fastqc.zip\n",
            "    log: logs/fastqc/trimmed/SRR9879594.log\n",
            "    jobid: 75\n",
            "    reason: Missing output files: results/fastqc/trimmed/SRR9879594_trimmed_fastqc.zip; Input files updated by another job: data/trimmed/SRR9879594.trimmed.fastq\n",
            "    wildcards: srr=SRR9879594\n",
            "    threads: 2\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule hisat2_align:\n",
            "    input: data/trimmed/SRR987961.trimmed.fastq, resources/hisat2_index\n",
            "    output: results/aligned/SRR987961.sam\n",
            "    log: logs/hisat2/SRR987961.log\n",
            "    jobid: 39\n",
            "    reason: Missing output files: results/aligned/SRR987961.sam; Input files updated by another job: resources/hisat2_index, data/trimmed/SRR987961.trimmed.fastq\n",
            "    wildcards: srr=SRR987961\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule hisat2_align:\n",
            "    input: data/trimmed/SRR9879599.trimmed.fastq, resources/hisat2_index\n",
            "    output: results/aligned/SRR9879599.sam\n",
            "    log: logs/hisat2/SRR9879599.log\n",
            "    jobid: 29\n",
            "    reason: Missing output files: results/aligned/SRR9879599.sam; Input files updated by another job: data/trimmed/SRR9879599.trimmed.fastq, resources/hisat2_index\n",
            "    wildcards: srr=SRR9879599\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule samtools_sort:\n",
            "    input: results/aligned/SRR987960.sam\n",
            "    output: results/aligned/SRR987960.sorted.bam\n",
            "    log: logs/samtools/SRR987960.sort.log\n",
            "    jobid: 33\n",
            "    reason: Missing output files: results/aligned/SRR987960.sorted.bam; Input files updated by another job: results/aligned/SRR987960.sam\n",
            "    wildcards: srr=SRR987960\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule samtools_sort:\n",
            "    input: results/aligned/SRR987962.sam\n",
            "    output: results/aligned/SRR987962.sorted.bam\n",
            "    log: logs/samtools/SRR987962.sort.log\n",
            "    jobid: 43\n",
            "    reason: Missing output files: results/aligned/SRR987962.sorted.bam; Input files updated by another job: results/aligned/SRR987962.sam\n",
            "    wildcards: srr=SRR987962\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule samtools_sort:\n",
            "    input: results/aligned/SRR9879597.sam\n",
            "    output: results/aligned/SRR9879597.sorted.bam\n",
            "    log: logs/samtools/SRR9879597.sort.log\n",
            "    jobid: 18\n",
            "    reason: Missing output files: results/aligned/SRR9879597.sorted.bam; Input files updated by another job: results/aligned/SRR9879597.sam\n",
            "    wildcards: srr=SRR9879597\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule samtools_sort:\n",
            "    input: results/aligned/SRR987964.sam\n",
            "    output: results/aligned/SRR987964.sorted.bam\n",
            "    log: logs/samtools/SRR987964.sort.log\n",
            "    jobid: 53\n",
            "    reason: Missing output files: results/aligned/SRR987964.sorted.bam; Input files updated by another job: results/aligned/SRR987964.sam\n",
            "    wildcards: srr=SRR987964\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule samtools_sort:\n",
            "    input: results/aligned/SRR9879598.sam\n",
            "    output: results/aligned/SRR9879598.sorted.bam\n",
            "    log: logs/samtools/SRR9879598.sort.log\n",
            "    jobid: 23\n",
            "    reason: Missing output files: results/aligned/SRR9879598.sorted.bam; Input files updated by another job: results/aligned/SRR9879598.sam\n",
            "    wildcards: srr=SRR9879598\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule samtools_sort:\n",
            "    input: results/aligned/SRR9879594.sam\n",
            "    output: results/aligned/SRR9879594.sorted.bam\n",
            "    log: logs/samtools/SRR9879594.sort.log\n",
            "    jobid: 2\n",
            "    reason: Missing output files: results/aligned/SRR9879594.sorted.bam; Input files updated by another job: results/aligned/SRR9879594.sam\n",
            "    wildcards: srr=SRR9879594\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule samtools_sort:\n",
            "    input: results/aligned/SRR9879595.sam\n",
            "    output: results/aligned/SRR9879595.sorted.bam\n",
            "    log: logs/samtools/SRR9879595.sort.log\n",
            "    jobid: 8\n",
            "    reason: Missing output files: results/aligned/SRR9879595.sorted.bam; Input files updated by another job: results/aligned/SRR9879595.sam\n",
            "    wildcards: srr=SRR9879595\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule samtools_sort:\n",
            "    input: results/aligned/SRR987963.sam\n",
            "    output: results/aligned/SRR987963.sorted.bam\n",
            "    log: logs/samtools/SRR987963.sort.log\n",
            "    jobid: 48\n",
            "    reason: Missing output files: results/aligned/SRR987963.sorted.bam; Input files updated by another job: results/aligned/SRR987963.sam\n",
            "    wildcards: srr=SRR987963\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule samtools_sort:\n",
            "    input: results/aligned/SRR987965.sam\n",
            "    output: results/aligned/SRR987965.sorted.bam\n",
            "    log: logs/samtools/SRR987965.sort.log\n",
            "    jobid: 58\n",
            "    reason: Missing output files: results/aligned/SRR987965.sorted.bam; Input files updated by another job: results/aligned/SRR987965.sam\n",
            "    wildcards: srr=SRR987965\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule samtools_sort:\n",
            "    input: results/aligned/SRR9879599.sam\n",
            "    output: results/aligned/SRR9879599.sorted.bam\n",
            "    log: logs/samtools/SRR9879599.sort.log\n",
            "    jobid: 28\n",
            "    reason: Missing output files: results/aligned/SRR9879599.sorted.bam; Input files updated by another job: results/aligned/SRR9879599.sam\n",
            "    wildcards: srr=SRR9879599\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule samtools_sort:\n",
            "    input: results/aligned/SRR9879596.sam\n",
            "    output: results/aligned/SRR9879596.sorted.bam\n",
            "    log: logs/samtools/SRR9879596.sort.log\n",
            "    jobid: 13\n",
            "    reason: Missing output files: results/aligned/SRR9879596.sorted.bam; Input files updated by another job: results/aligned/SRR9879596.sam\n",
            "    wildcards: srr=SRR9879596\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule samtools_sort:\n",
            "    input: results/aligned/SRR987961.sam\n",
            "    output: results/aligned/SRR987961.sorted.bam\n",
            "    log: logs/samtools/SRR987961.sort.log\n",
            "    jobid: 38\n",
            "    reason: Missing output files: results/aligned/SRR987961.sorted.bam; Input files updated by another job: results/aligned/SRR987961.sam\n",
            "    wildcards: srr=SRR987961\n",
            "    threads: 4\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule htseq_count:\n",
            "    input: results/aligned/SRR9879599.sorted.bam, resources/Solanum_lycopersicum.SL3.0.61.chr.gff3\n",
            "    output: results/counts/SRR9879599.counts.txt\n",
            "    log: logs/htseq/SRR9879599.log\n",
            "    jobid: 27\n",
            "    reason: Missing output files: results/counts/SRR9879599.counts.txt; Input files updated by another job: results/aligned/SRR9879599.sorted.bam\n",
            "    wildcards: srr=SRR9879599\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule htseq_count:\n",
            "    input: results/aligned/SRR9879596.sorted.bam, resources/Solanum_lycopersicum.SL3.0.61.chr.gff3\n",
            "    output: results/counts/SRR9879596.counts.txt\n",
            "    log: logs/htseq/SRR9879596.log\n",
            "    jobid: 12\n",
            "    reason: Missing output files: results/counts/SRR9879596.counts.txt; Input files updated by another job: results/aligned/SRR9879596.sorted.bam\n",
            "    wildcards: srr=SRR9879596\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule htseq_count:\n",
            "    input: results/aligned/SRR9879598.sorted.bam, resources/Solanum_lycopersicum.SL3.0.61.chr.gff3\n",
            "    output: results/counts/SRR9879598.counts.txt\n",
            "    log: logs/htseq/SRR9879598.log\n",
            "    jobid: 22\n",
            "    reason: Missing output files: results/counts/SRR9879598.counts.txt; Input files updated by another job: results/aligned/SRR9879598.sorted.bam\n",
            "    wildcards: srr=SRR9879598\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule htseq_count:\n",
            "    input: results/aligned/SRR987965.sorted.bam, resources/Solanum_lycopersicum.SL3.0.61.chr.gff3\n",
            "    output: results/counts/SRR987965.counts.txt\n",
            "    log: logs/htseq/SRR987965.log\n",
            "    jobid: 57\n",
            "    reason: Missing output files: results/counts/SRR987965.counts.txt; Input files updated by another job: results/aligned/SRR987965.sorted.bam\n",
            "    wildcards: srr=SRR987965\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule htseq_count:\n",
            "    input: results/aligned/SRR987961.sorted.bam, resources/Solanum_lycopersicum.SL3.0.61.chr.gff3\n",
            "    output: results/counts/SRR987961.counts.txt\n",
            "    log: logs/htseq/SRR987961.log\n",
            "    jobid: 37\n",
            "    reason: Missing output files: results/counts/SRR987961.counts.txt; Input files updated by another job: results/aligned/SRR987961.sorted.bam\n",
            "    wildcards: srr=SRR987961\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule htseq_count:\n",
            "    input: results/aligned/SRR9879597.sorted.bam, resources/Solanum_lycopersicum.SL3.0.61.chr.gff3\n",
            "    output: results/counts/SRR9879597.counts.txt\n",
            "    log: logs/htseq/SRR9879597.log\n",
            "    jobid: 17\n",
            "    reason: Missing output files: results/counts/SRR9879597.counts.txt; Input files updated by another job: results/aligned/SRR9879597.sorted.bam\n",
            "    wildcards: srr=SRR9879597\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule htseq_count:\n",
            "    input: results/aligned/SRR9879594.sorted.bam, resources/Solanum_lycopersicum.SL3.0.61.chr.gff3\n",
            "    output: results/counts/SRR9879594.counts.txt\n",
            "    log: logs/htseq/SRR9879594.log\n",
            "    jobid: 1\n",
            "    reason: Missing output files: results/counts/SRR9879594.counts.txt; Input files updated by another job: results/aligned/SRR9879594.sorted.bam\n",
            "    wildcards: srr=SRR9879594\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule multiqc:\n",
            "    input: results/fastqc/raw/SRR9879594_fastqc.zip, results/fastqc/raw/SRR9879595_fastqc.zip, results/fastqc/raw/SRR9879596_fastqc.zip, results/fastqc/raw/SRR9879597_fastqc.zip, results/fastqc/raw/SRR9879598_fastqc.zip, results/fastqc/raw/SRR9879599_fastqc.zip, results/fastqc/raw/SRR987960_fastqc.zip, results/fastqc/raw/SRR987961_fastqc.zip, results/fastqc/raw/SRR987962_fastqc.zip, results/fastqc/raw/SRR987963_fastqc.zip, results/fastqc/raw/SRR987964_fastqc.zip, results/fastqc/raw/SRR987965_fastqc.zip, results/fastqc/trimmed/SRR9879594_trimmed_fastqc.zip, results/fastqc/trimmed/SRR9879595_trimmed_fastqc.zip, results/fastqc/trimmed/SRR9879596_trimmed_fastqc.zip, results/fastqc/trimmed/SRR9879597_trimmed_fastqc.zip, results/fastqc/trimmed/SRR9879598_trimmed_fastqc.zip, results/fastqc/trimmed/SRR9879599_trimmed_fastqc.zip, results/fastqc/trimmed/SRR987960_trimmed_fastqc.zip, results/fastqc/trimmed/SRR987961_trimmed_fastqc.zip, results/fastqc/trimmed/SRR987962_trimmed_fastqc.zip, results/fastqc/trimmed/SRR987963_trimmed_fastqc.zip, results/fastqc/trimmed/SRR987964_trimmed_fastqc.zip, results/fastqc/trimmed/SRR987965_trimmed_fastqc.zip, logs/trimmomatic/SRR9879594.log, logs/trimmomatic/SRR9879595.log, logs/trimmomatic/SRR9879596.log, logs/trimmomatic/SRR9879597.log, logs/trimmomatic/SRR9879598.log, logs/trimmomatic/SRR9879599.log, logs/trimmomatic/SRR987960.log, logs/trimmomatic/SRR987961.log, logs/trimmomatic/SRR987962.log, logs/trimmomatic/SRR987963.log, logs/trimmomatic/SRR987964.log, logs/trimmomatic/SRR987965.log, results/aligned/SRR9879594.sorted.bam, results/aligned/SRR9879595.sorted.bam, results/aligned/SRR9879596.sorted.bam, results/aligned/SRR9879597.sorted.bam, results/aligned/SRR9879598.sorted.bam, results/aligned/SRR9879599.sorted.bam, results/aligned/SRR987960.sorted.bam, results/aligned/SRR987961.sorted.bam, results/aligned/SRR987962.sorted.bam, results/aligned/SRR987963.sorted.bam, results/aligned/SRR987964.sorted.bam, results/aligned/SRR987965.sorted.bam\n",
            "    output: multiqc_report.html\n",
            "    jobid: 62\n",
            "    reason: Missing output files: multiqc_report.html; Input files updated by another job: results/aligned/SRR987964.sorted.bam, results/fastqc/raw/SRR987963_fastqc.zip, results/fastqc/raw/SRR9879594_fastqc.zip, results/fastqc/raw/SRR9879596_fastqc.zip, results/fastqc/raw/SRR987964_fastqc.zip, results/fastqc/trimmed/SRR9879598_trimmed_fastqc.zip, results/fastqc/trimmed/SRR987965_trimmed_fastqc.zip, results/aligned/SRR987963.sorted.bam, results/fastqc/trimmed/SRR987963_trimmed_fastqc.zip, logs/trimmomatic/SRR9879594.log, results/fastqc/trimmed/SRR9879596_trimmed_fastqc.zip, logs/trimmomatic/SRR987960.log, logs/trimmomatic/SRR987963.log, logs/trimmomatic/SRR9879598.log, results/fastqc/trimmed/SRR9879595_trimmed_fastqc.zip, results/aligned/SRR9879595.sorted.bam, results/aligned/SRR987965.sorted.bam, logs/trimmomatic/SRR9879599.log, results/fastqc/trimmed/SRR9879599_trimmed_fastqc.zip, logs/trimmomatic/SRR987962.log, results/aligned/SRR9879597.sorted.bam, results/fastqc/raw/SRR987965_fastqc.zip, results/fastqc/raw/SRR987960_fastqc.zip, logs/trimmomatic/SRR987964.log, results/aligned/SRR987960.sorted.bam, results/fastqc/trimmed/SRR987960_trimmed_fastqc.zip, results/fastqc/raw/SRR9879595_fastqc.zip, results/fastqc/trimmed/SRR9879597_trimmed_fastqc.zip, results/aligned/SRR987961.sorted.bam, results/aligned/SRR9879596.sorted.bam, results/fastqc/raw/SRR9879598_fastqc.zip, results/fastqc/raw/SRR987961_fastqc.zip, results/fastqc/trimmed/SRR987961_trimmed_fastqc.zip, results/aligned/SRR9879598.sorted.bam, logs/trimmomatic/SRR987965.log, results/fastqc/raw/SRR987962_fastqc.zip, results/fastqc/raw/SRR9879597_fastqc.zip, results/fastqc/trimmed/SRR987962_trimmed_fastqc.zip, results/fastqc/trimmed/SRR987964_trimmed_fastqc.zip, logs/trimmomatic/SRR9879595.log, results/aligned/SRR9879594.sorted.bam, logs/trimmomatic/SRR9879596.log, results/aligned/SRR987962.sorted.bam, results/fastqc/trimmed/SRR9879594_trimmed_fastqc.zip, logs/trimmomatic/SRR9879597.log, results/aligned/SRR9879599.sorted.bam, results/fastqc/raw/SRR9879599_fastqc.zip, logs/trimmomatic/SRR987961.log\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule htseq_count:\n",
            "    input: results/aligned/SRR9879595.sorted.bam, resources/Solanum_lycopersicum.SL3.0.61.chr.gff3\n",
            "    output: results/counts/SRR9879595.counts.txt\n",
            "    log: logs/htseq/SRR9879595.log\n",
            "    jobid: 7\n",
            "    reason: Missing output files: results/counts/SRR9879595.counts.txt; Input files updated by another job: results/aligned/SRR9879595.sorted.bam\n",
            "    wildcards: srr=SRR9879595\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule htseq_count:\n",
            "    input: results/aligned/SRR987963.sorted.bam, resources/Solanum_lycopersicum.SL3.0.61.chr.gff3\n",
            "    output: results/counts/SRR987963.counts.txt\n",
            "    log: logs/htseq/SRR987963.log\n",
            "    jobid: 47\n",
            "    reason: Missing output files: results/counts/SRR987963.counts.txt; Input files updated by another job: results/aligned/SRR987963.sorted.bam\n",
            "    wildcards: srr=SRR987963\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule htseq_count:\n",
            "    input: results/aligned/SRR987964.sorted.bam, resources/Solanum_lycopersicum.SL3.0.61.chr.gff3\n",
            "    output: results/counts/SRR987964.counts.txt\n",
            "    log: logs/htseq/SRR987964.log\n",
            "    jobid: 52\n",
            "    reason: Missing output files: results/counts/SRR987964.counts.txt; Input files updated by another job: results/aligned/SRR987964.sorted.bam\n",
            "    wildcards: srr=SRR987964\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule htseq_count:\n",
            "    input: results/aligned/SRR987960.sorted.bam, resources/Solanum_lycopersicum.SL3.0.61.chr.gff3\n",
            "    output: results/counts/SRR987960.counts.txt\n",
            "    log: logs/htseq/SRR987960.log\n",
            "    jobid: 32\n",
            "    reason: Missing output files: results/counts/SRR987960.counts.txt; Input files updated by another job: results/aligned/SRR987960.sorted.bam\n",
            "    wildcards: srr=SRR987960\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule htseq_count:\n",
            "    input: results/aligned/SRR987962.sorted.bam, resources/Solanum_lycopersicum.SL3.0.61.chr.gff3\n",
            "    output: results/counts/SRR987962.counts.txt\n",
            "    log: logs/htseq/SRR987962.log\n",
            "    jobid: 42\n",
            "    reason: Missing output files: results/counts/SRR987962.counts.txt; Input files updated by another job: results/aligned/SRR987962.sorted.bam\n",
            "    wildcards: srr=SRR987962\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 06:40:47 2025]\n",
            "rule all:\n",
            "    input: results/counts/SRR9879594.counts.txt, results/counts/SRR9879595.counts.txt, results/counts/SRR9879596.counts.txt, results/counts/SRR9879597.counts.txt, results/counts/SRR9879598.counts.txt, results/counts/SRR9879599.counts.txt, results/counts/SRR987960.counts.txt, results/counts/SRR987961.counts.txt, results/counts/SRR987962.counts.txt, results/counts/SRR987963.counts.txt, results/counts/SRR987964.counts.txt, results/counts/SRR987965.counts.txt, multiqc_report.html\n",
            "    jobid: 0\n",
            "    reason: Input files updated by another job: results/counts/SRR9879599.counts.txt, results/counts/SRR9879596.counts.txt, multiqc_report.html, results/counts/SRR9879594.counts.txt, results/counts/SRR9879598.counts.txt, results/counts/SRR987964.counts.txt, results/counts/SRR9879595.counts.txt, results/counts/SRR9879597.counts.txt, results/counts/SRR987962.counts.txt, results/counts/SRR987965.counts.txt, results/counts/SRR987961.counts.txt, results/counts/SRR987960.counts.txt, results/counts/SRR987963.counts.txt\n",
            "    resources: tmpdir=<TBD>\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\u001b[33mJob stats:\n",
            "job               count\n",
            "--------------  -------\n",
            "all                   1\n",
            "download_sra         12\n",
            "fastqc_raw           12\n",
            "fastqc_trimmed       12\n",
            "hisat2_align         12\n",
            "hisat2_index          1\n",
            "htseq_count          12\n",
            "multiqc               1\n",
            "samtools_sort        12\n",
            "trim                 12\n",
            "total                87\n",
            "\u001b[0m\n",
            "\u001b[33mReasons:\n",
            "    (check individual jobs above for details)\n",
            "    input files updated by another job:\n",
            "        all, fastqc_raw, fastqc_trimmed, hisat2_align, htseq_count, multiqc, samtools_sort, trim\n",
            "    output files have to be generated:\n",
            "        download_sra, fastqc_raw, fastqc_trimmed, hisat2_align, hisat2_index, htseq_count, multiqc, samtools_sort, trim\u001b[0m\n",
            "\u001b[33mThis was a dry-run (flag -n). The order of jobs does not reflect the order of execution.\u001b[0m\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!snakemake --cores 8 hisat2_index"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Lb5PtWrnlGgz",
        "outputId": "ac739534-d24f-457c-db65-04cc6d14eaa9"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "The flag 'directory' used in rule hisat2_align is only valid for outputs, not inputs.\n",
            "\u001b[33mAssuming unrestricted shared filesystem usage.\u001b[0m\n",
            "\u001b[33mhost: bb74ec2f9ae5\u001b[0m\n",
            "\u001b[33mBuilding DAG of jobs...\u001b[0m\n",
            "\u001b[33mUsing shell: /usr/bin/bash\u001b[0m\n",
            "\u001b[32mProvided cores: 8\u001b[0m\n",
            "\u001b[32mRules claiming more threads will be scaled down.\u001b[0m\n",
            "\u001b[33mConda environments: ignored\u001b[0m\n",
            "\u001b[33mJob stats:\n",
            "job             count\n",
            "------------  -------\n",
            "hisat2_index        1\n",
            "total               1\n",
            "\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 07:15:09 2025]\n",
            "localrule hisat2_index:\n",
            "    input: resources/Solanum_lycopersicum.SL3.0.dna.toplevel.fa\n",
            "    output: resources/hisat2_index\n",
            "    log: logs/hisat2/index.log\n",
            "    jobid: 0\n",
            "    reason: Missing output files: resources/hisat2_index\n",
            "    threads: 8\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "\u001b[33mWaiting at most 5 seconds for missing files:\n",
            "resources/hisat2_index (missing locally)\u001b[0m\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!snakemake --cores 4 --use-conda"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "a5wombBZnwWe",
        "outputId": "340406e6-980d-4544-fc3e-9a9ec44fc5f7"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "The flag 'directory' used in rule hisat2_align is only valid for outputs, not inputs.\n",
            "\u001b[33mAssuming unrestricted shared filesystem usage.\u001b[0m\n",
            "\u001b[33mhost: bb74ec2f9ae5\u001b[0m\n",
            "\u001b[33mBuilding DAG of jobs...\u001b[0m\n",
            "\u001b[33mYour conda installation is not configured to use strict channel priorities. This is however important for having robust and correct environments (for details, see https://conda-forge.org/docs/user/tipsandtricks.html). Please consider to configure strict priorities by executing 'conda config --set channel_priority strict'.\u001b[0m\n",
            "\u001b[33mUsing shell: /usr/bin/bash\u001b[0m\n",
            "\u001b[32mProvided cores: 4\u001b[0m\n",
            "\u001b[32mRules claiming more threads will be scaled down.\u001b[0m\n",
            "\u001b[33mJob stats:\n",
            "job               count\n",
            "--------------  -------\n",
            "all                   1\n",
            "download_sra         12\n",
            "fastqc_raw           12\n",
            "fastqc_trimmed       12\n",
            "hisat2_align         12\n",
            "hisat2_index          1\n",
            "htseq_count          12\n",
            "multiqc               1\n",
            "samtools_sort        12\n",
            "trim                 12\n",
            "total                87\n",
            "\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\n",
            "\u001b[32m[Tue May 27 07:01:48 2025]\n",
            "localrule hisat2_index:\n",
            "    input: resources/Solanum_lycopersicum.SL3.0.dna.toplevel.fa\n",
            "    output: resources/hisat2_index\n",
            "    log: logs/hisat2/index.log\n",
            "    jobid: 6\n",
            "    reason: Missing output files: resources/hisat2_index\n",
            "    threads: 4\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[33m\u001b[0m\n",
            "Hit:1 http://security.ubuntu.com/ubuntu jammy-security InRelease\n",
            "Hit:2 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease\n",
            "Hit:3 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease\n",
            "Hit:4 https://r2u.stat.illinois.edu/ubuntu jammy InRelease\n",
            "Hit:5 http://archive.ubuntu.com/ubuntu jammy InRelease\n",
            "Hit:6 http://archive.ubuntu.com/ubuntu jammy-updates InRelease\n",
            "Hit:7 http://archive.ubuntu.com/ubuntu jammy-backports InRelease\n",
            "Hit:8 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy InRelease\n",
            "Hit:9 https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy InRelease\n",
            "Hit:10 https://ppa.launchpadcontent.net/ubuntugis/ppa/ubuntu jammy InRelease\n",
            "0% [Working]\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "Reading package lists... Done\n",
            "W: Skipping acquire of configured file 'main/source/Sources' as repository 'https://r2u.stat.illinois.edu/ubuntu jammy InRelease' does not seem to provide it (sources.list entry misspelt?)\n",
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "hisat2 is already the newest version (2.2.1-3).\n",
            "0 upgraded, 0 newly installed, 0 to remove and 100 not upgraded.\n",
            "\u001b[31mRuleException:\n",
            "CalledProcessError in file \"/content/Snakefile\", line 86:\n",
            "Command 'set -euo pipefail;  \n",
            "        mkdir -p logs/hisat2\n",
            "        sudo apt-get update\n",
            "        sudo apt-get -y install hisat2\n",
            "        hisat2-build -p 4 resources/Solanum_lycopersicum.SL3.0.dna.toplevel.fa resources/hisat2_index/tomato reference_index > logs/hisat2/index.log 2>&1' returned non-zero exit status 1.\u001b[0m\n",
            "\u001b[31m[Tue May 27 07:01:56 2025]\n",
            "Error in rule hisat2_index:\n",
            "    message: None\n",
            "    jobid: 6\n",
            "    input: resources/Solanum_lycopersicum.SL3.0.dna.toplevel.fa\n",
            "    output: resources/hisat2_index\n",
            "    log: logs/hisat2/index.log (check log file(s) for error details)\n",
            "    shell:\n",
            "        \n",
            "        mkdir -p logs/hisat2\n",
            "        sudo apt-get update\n",
            "        sudo apt-get -y install hisat2\n",
            "        hisat2-build -p 4 resources/Solanum_lycopersicum.SL3.0.dna.toplevel.fa resources/hisat2_index/tomato reference_index > logs/hisat2/index.log 2>&1\n",
            "        \n",
            "        (command exited with non-zero exit code)\u001b[0m\n",
            "\u001b[33mShutting down, this might take some time.\u001b[0m\n",
            "\u001b[31mExiting because a job execution failed. Look below for error messages\u001b[0m\n",
            "\u001b[31m[Tue May 27 07:01:56 2025]\n",
            "Error in rule hisat2_index:\n",
            "    message: None\n",
            "    jobid: 6\n",
            "    input: resources/Solanum_lycopersicum.SL3.0.dna.toplevel.fa\n",
            "    output: resources/hisat2_index\n",
            "    log: logs/hisat2/index.log (check log file(s) for error details)\n",
            "    shell:\n",
            "        \n",
            "        mkdir -p logs/hisat2\n",
            "        sudo apt-get update\n",
            "        sudo apt-get -y install hisat2\n",
            "        hisat2-build -p 4 resources/Solanum_lycopersicum.SL3.0.dna.toplevel.fa resources/hisat2_index/tomato reference_index > logs/hisat2/index.log 2>&1\n",
            "        \n",
            "        (command exited with non-zero exit code)\u001b[0m\n",
            "\u001b[33mComplete log(s): /content/.snakemake/log/2025-05-27T070145.099762.snakemake.log\u001b[0m\n",
            "\u001b[31mWorkflowError:\n",
            "At least one job did not complete successfully.\u001b[0m\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!snakemake -s Snakefile_pipeline_1 -j 15\n",
        "!snakemake -s Snakefile_pipeline_2 -j 40\n",
        "!snakemake -s Snakefile_pipeline_3 -j 10\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ExPyWuKwZBHl",
        "outputId": "a278e3a0-5148-4b6a-c01e-d6bc05249576"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Error: Snakefile \"Snakefile_pipeline_1\" not found.\n",
            "Error: Snakefile \"Snakefile_pipeline_2\" not found.\n",
            "Error: Snakefile \"Snakefile_pipeline_3\" not found.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!touch snakefile3"
      ],
      "metadata": {
        "id": "f5RQaZQONQFn"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!conda install -c bioconda htseq"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "nVND2ULrGaMi",
        "outputId": "c835ba99-586a-4c10-d215-1385aa26ab2d"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Channels:\n",
            " - bioconda\n",
            " - defaults\n",
            "Platform: linux-64\n",
            "Collecting package metadata (repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "Solving environment: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bfailed\n",
            "\n",
            "LibMambaUnsatisfiableError: Encountered problems while solving:\n",
            "  - nothing provides python_abi 3.6.* *_cp36m needed by htseq-0.12.3-py36hb3f55d8_0\n",
            "\n",
            "Could not solve for environment specs\n",
            "The following packages are incompatible\n",
            "├─ \u001b[32mhtseq =* *\u001b[0m is installable with the potential options\n",
            "│  ├─ \u001b[32mhtseq [0.11.0|0.11.1|...|0.9.1]\u001b[0m would require\n",
            "│  │  └─ \u001b[32mpython [=2.7 *|>=2.7,<2.8.0a0 *]\u001b[0m, which can be installed;\n",
            "│  ├─ \u001b[32mhtseq [0.11.0|0.11.1|0.11.2|0.7.2|0.9.1]\u001b[0m would require\n",
            "│  │  └─ \u001b[32mpython [=3.5 *|>=3.5,<3.6.0a0 *]\u001b[0m, which can be installed;\n",
            "│  ├─ \u001b[32mhtseq [0.11.0|0.11.1|...|0.9.1]\u001b[0m would require\n",
            "│  │  └─ \u001b[32mpython [=3.6 *|>=3.6,<3.7.0a0 *]\u001b[0m, which can be installed;\n",
            "│  ├─ \u001b[32mhtseq [0.11.2|0.11.3]\u001b[0m would require\n",
            "│  │  └─ \u001b[32mpython >=3.7,<3.8.0a0 *\u001b[0m, which can be installed;\n",
            "│  ├─ \u001b[31mhtseq [0.12.3|0.12.4|0.13.5|0.7.2|0.9.1]\u001b[0m would require\n",
            "│  │  └─ \u001b[31mpython_abi =3.6 *_cp36m\u001b[0m, which does not exist (perhaps a missing channel);\n",
            "│  ├─ \u001b[31mhtseq [0.12.3|0.12.4|...|2.0.1]\u001b[0m would require\n",
            "│  │  └─ \u001b[31mpython_abi =3.7 *_cp37m\u001b[0m, which does not exist (perhaps a missing channel);\n",
            "│  ├─ \u001b[31mhtseq [0.12.4|0.13.5|...|2.0.1]\u001b[0m would require\n",
            "│  │  └─ \u001b[31mpython_abi =3.8 *_cp38\u001b[0m, which does not exist (perhaps a missing channel);\n",
            "│  ├─ \u001b[31mhtseq [0.13.5|1.99.2|2.0.1]\u001b[0m would require\n",
            "│  │  └─ \u001b[31mpython_abi =3.9 *_cp39\u001b[0m, which does not exist (perhaps a missing channel);\n",
            "│  ├─ \u001b[31mhtseq 0.6.1\u001b[0m would require\n",
            "│  │  └─ \u001b[31mmatplotlib ==1.2.1 *\u001b[0m, which does not exist (perhaps a missing channel);\n",
            "│  ├─ \u001b[31mhtseq [0.6.1|0.6.1.post1|0.7.2|0.9.1]\u001b[0m would require\n",
            "│  │  └─ \u001b[31mpython_abi =2.7 *_cp27mu\u001b[0m, which does not exist (perhaps a missing channel);\n",
            "│  ├─ \u001b[31mhtseq 0.7.2\u001b[0m would require\n",
            "│  │  └─ \u001b[31mpython =3.4 *\u001b[0m, which does not exist (perhaps a missing channel);\n",
            "│  ├─ \u001b[31mhtseq [2.0.2|2.0.3|2.0.4|2.0.5]\u001b[0m would require\n",
            "│  │  └─ \u001b[31mlibgcc-ng >=12 *\u001b[0m, which does not exist (perhaps a missing channel);\n",
            "│  └─ \u001b[31mhtseq 2.0.5\u001b[0m would require\n",
            "│     └─ \u001b[31mlibgcc >=13 *\u001b[0m, which does not exist (perhaps a missing channel);\n",
            "└─ \u001b[31mpin on python 3.13.* =* *\u001b[0m is not installable because it requires\n",
            "   └─ \u001b[31mpython =3.13 *\u001b[0m, which conflicts with any installable versions previously reported.\n",
            "\n",
            "Pins seem to be involved in the conflict. Currently pinned specs:\n",
            " - python=3.13\n",
            "\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "SRR_IDS = [f\"SRR98795{i:02d}\" for i in range(94, 106)]  # SRR9879594 to SRR9879605\n",
        "print(SRR_IDS)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "FuB_5Ew34F4V",
        "outputId": "8bdbfce9-e40c-4b12-f215-3b098bff56f6"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "['SRR9879594', 'SRR9879595', 'SRR9879596', 'SRR9879597', 'SRR9879598', 'SRR9879599', 'SRR98795100', 'SRR98795101', 'SRR98795102', 'SRR98795103', 'SRR98795104', 'SRR98795105']\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "SRR_IDS = [f\"SRR987959{i}\" for i in range(4, 10)] + [f\"SRR98796{i}\" for i in range(0, 6)]\n",
        "print(SRA_IDS)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "TO_CRdlt4Wsl",
        "outputId": "b0bc60ae-ea98-4878-d0de-5083643600f6"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "['SRR9879594', 'SRR9879595', 'SRR9879596', 'SRR9879597', 'SRR9879598', 'SRR9879599', 'SRR987960', 'SRR987961', 'SRR987962', 'SRR987963', 'SRR987964', 'SRR987965']\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://cloud.biohpc.swmed.edu/index.php/s/hisat2-220-download/download -O hisat2.zip\n",
        "!unzip hisat2.zip"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "NWxpuC2aPi9i",
        "outputId": "ee2eef26-9a6c-487f-d866-f5dea4e8f367"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2025-05-26 20:15:10--  https://cloud.biohpc.swmed.edu/index.php/s/hisat2-220-download/download\n",
            "Resolving cloud.biohpc.swmed.edu (cloud.biohpc.swmed.edu)... 129.112.9.92\n",
            "Connecting to cloud.biohpc.swmed.edu (cloud.biohpc.swmed.edu)|129.112.9.92|:443... connected.\n",
            "HTTP request sent, awaiting response... 404 Not Found\n",
            "2025-05-26 20:15:11 ERROR 404: Not Found.\n",
            "\n",
            "Archive:  hisat2.zip\n",
            "  End-of-central-directory signature not found.  Either this file is not\n",
            "  a zipfile, or it constitutes one disk of a multi-part archive.  In the\n",
            "  latter case the central directory and zipfile comment will be found on\n",
            "  the last disk(s) of this archive.\n",
            "unzip:  cannot find zipfile directory in one of hisat2.zip or\n",
            "        hisat2.zip.zip, and cannot find hisat2.zip.ZIP, period.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!conda install -c bioconda hisat2 -y"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "5TkZ5oiUPm1g",
        "outputId": "4a0766ad-58b6-42e1-dd1d-c90574cc8217"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Channels:\n",
            " - bioconda\n",
            " - defaults\n",
            "Platform: linux-64\n",
            "Collecting package metadata (repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "Solving environment: / \b\b- \b\b\\ \b\bdone\n",
            "\n",
            "# All requested packages already installed.\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!sudo apt-get update\n",
        "!sudo apt-get -y install hisat2"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "cNP4wkP1P4Br",
        "outputId": "a4ff3fa0-59e5-4cf7-df95-32901f3d14f0"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\r0% [Working]\r            \rGet:1 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease [1,581 B]\n",
            "\r0% [Connecting to archive.ubuntu.com] [Connecting to security.ubuntu.com (185.1\r0% [Connecting to archive.ubuntu.com] [Connecting to security.ubuntu.com (185.1\r                                                                               \rGet:2 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease [3,632 B]\n",
            "\r0% [Connecting to archive.ubuntu.com (91.189.91.83)] [Waiting for headers] [Wai\r                                                                               \rHit:3 http://archive.ubuntu.com/ubuntu jammy InRelease\n",
            "\r0% [Waiting for headers] [Waiting for headers] [Waiting for headers] [Connected\r                                                                               \rGet:4 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [128 kB]\n",
            "\r0% [4 InRelease 12.7 kB/128 kB 10%] [Waiting for headers] [Waiting for headers]\r                                                                               \rGet:5 https://r2u.stat.illinois.edu/ubuntu jammy InRelease [6,555 B]\n",
            "\r0% [4 InRelease 73.5 kB/128 kB 57%] [Waiting for headers] [5 InRelease 6,555 B/\r0% [4 InRelease 77.9 kB/128 kB 61%] [Waiting for headers] [Connected to ppa.lau\r0% [Waiting for headers] [Connected to ppa.launchpadcontent.net (185.125.190.80\r                                                                               \rGet:6 http://security.ubuntu.com/ubuntu jammy-security InRelease [129 kB]\n",
            "Get:7 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [127 kB]\n",
            "Get:8 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  Packages [1,683 kB]\n",
            "Hit:9 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy InRelease\n",
            "Get:10 https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy InRelease [24.3 kB]\n",
            "Get:11 https://r2u.stat.illinois.edu/ubuntu jammy/main all Packages [8,969 kB]\n",
            "Get:12 https://ppa.launchpadcontent.net/ubuntugis/ppa/ubuntu jammy InRelease [24.6 kB]\n",
            "Get:13 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages [3,255 kB]\n",
            "Get:14 http://archive.ubuntu.com/ubuntu jammy-updates/restricted amd64 Packages [4,540 kB]\n",
            "Get:15 https://r2u.stat.illinois.edu/ubuntu jammy/main amd64 Packages [2,729 kB]\n",
            "Get:16 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages [1,552 kB]\n",
            "Get:17 http://archive.ubuntu.com/ubuntu jammy-backports/universe amd64 Packages [35.2 kB]\n",
            "Get:18 http://archive.ubuntu.com/ubuntu jammy-backports/main amd64 Packages [83.2 kB]\n",
            "Get:19 http://security.ubuntu.com/ubuntu jammy-security/main amd64 Packages [2,944 kB]\n",
            "Get:20 https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy/main amd64 Packages [46.6 kB]\n",
            "Get:21 https://ppa.launchpadcontent.net/ubuntugis/ppa/ubuntu jammy/main amd64 Packages [75.1 kB]\n",
            "Get:22 http://security.ubuntu.com/ubuntu jammy-security/restricted amd64 Packages [4,387 kB]\n",
            "Get:23 http://security.ubuntu.com/ubuntu jammy-security/universe amd64 Packages [1,245 kB]\n",
            "Fetched 32.0 MB in 4s (8,800 kB/s)\n",
            "Reading package lists... Done\n",
            "W: Skipping acquire of configured file 'main/source/Sources' as repository 'https://r2u.stat.illinois.edu/ubuntu jammy InRelease' does not seem to provide it (sources.list entry misspelt?)\n",
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  bcftools libhts3 libhtscodecs2 python3-hisat2 samtools\n",
            "Suggested packages:\n",
            "  python3-numpy python3-matplotlib texlive-latex-recommended cwltool\n",
            "The following NEW packages will be installed:\n",
            "  bcftools hisat2 libhts3 libhtscodecs2 python3-hisat2 samtools\n",
            "0 upgraded, 6 newly installed, 0 to remove and 100 not upgraded.\n",
            "Need to get 5,505 kB of archives.\n",
            "After this operation, 17.1 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libhtscodecs2 amd64 1.1.1-3 [53.2 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libhts3 amd64 1.13+ds-2build1 [390 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu jammy/universe amd64 bcftools amd64 1.13-1 [697 kB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu jammy/universe amd64 hisat2 amd64 2.2.1-3 [3,832 kB]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu jammy/universe amd64 python3-hisat2 all 2.2.1-3 [12.7 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu jammy/universe amd64 samtools amd64 1.13-4 [520 kB]\n",
            "Fetched 5,505 kB in 0s (27.4 MB/s)\n",
            "debconf: unable to initialize frontend: Dialog\n",
            "debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 78, <> line 6.)\n",
            "debconf: falling back to frontend: Readline\n",
            "debconf: unable to initialize frontend: Readline\n",
            "debconf: (This frontend requires a controlling tty.)\n",
            "debconf: falling back to frontend: Teletype\n",
            "dpkg-preconfigure: unable to re-open stdin: \n",
            "Selecting previously unselected package libhtscodecs2:amd64.\n",
            "(Reading database ... 126102 files and directories currently installed.)\n",
            "Preparing to unpack .../0-libhtscodecs2_1.1.1-3_amd64.deb ...\n",
            "Unpacking libhtscodecs2:amd64 (1.1.1-3) ...\n",
            "Selecting previously unselected package libhts3:amd64.\n",
            "Preparing to unpack .../1-libhts3_1.13+ds-2build1_amd64.deb ...\n",
            "Unpacking libhts3:amd64 (1.13+ds-2build1) ...\n",
            "Selecting previously unselected package bcftools.\n",
            "Preparing to unpack .../2-bcftools_1.13-1_amd64.deb ...\n",
            "Unpacking bcftools (1.13-1) ...\n",
            "Selecting previously unselected package hisat2.\n",
            "Preparing to unpack .../3-hisat2_2.2.1-3_amd64.deb ...\n",
            "Unpacking hisat2 (2.2.1-3) ...\n",
            "Selecting previously unselected package python3-hisat2.\n",
            "Preparing to unpack .../4-python3-hisat2_2.2.1-3_all.deb ...\n",
            "Unpacking python3-hisat2 (2.2.1-3) ...\n",
            "Selecting previously unselected package samtools.\n",
            "Preparing to unpack .../5-samtools_1.13-4_amd64.deb ...\n",
            "Unpacking samtools (1.13-4) ...\n",
            "Setting up libhtscodecs2:amd64 (1.1.1-3) ...\n",
            "Setting up libhts3:amd64 (1.13+ds-2build1) ...\n",
            "Setting up bcftools (1.13-1) ...\n",
            "Setting up samtools (1.13-4) ...\n",
            "Setting up hisat2 (2.2.1-3) ...\n",
            "Setting up python3-hisat2 (2.2.1-3) ...\n",
            "Processing triggers for man-db (2.10.2-1) ...\n",
            "Processing triggers for libc-bin (2.35-0ubuntu3.8) ...\n",
            "/sbin/ldconfig.real: /usr/local/lib/libumf.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtcm_debug.so.1 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libhwloc.so.15 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_5.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbb.so.12 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_level_zero.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_0.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtcm.so.1 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_opencl.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc_proxy.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_loader.so.0 is not a symbolic link\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!head resources/Solanum_lycopersicum.SL3.0.dna.toplevel.fa"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "4tbhFeDnf5rD",
        "outputId": "0b19e039-0572-42a7-820e-f3725dbc57ac"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            ">1 dna:chromosome chromosome:SL3.0:1:1:98455869:1 REF\n",
            "GGCTTGTCCGACCCATTTGGAAAGGTCAAACGAGCCTCAAAGCGAGCATACCCCTCATTT\n",
            "CGACGATTTTCGTGTGCTATAGCACACCATTTTTTGGGTGATCCGGATTCCGACGTCAAA\n",
            "AATTCCAAATTTTTTCGTGGTCGTCCGTCAAGACCTTGTCTATGCATACGGTTGGCCATC\n",
            "ACGGCTTGTCCGACCCATTTGGAAGGTCAAACGAGCCCCGAAGCGAGCATACCCCTCCAT\n",
            "TCGACGATTTTCGTGTGCTATAGCACACCATTTTTTGGGTGATCCGGATTCCGACGTCAA\n",
            "AAATGCAAAAAGATTTCGTGGACGTCCGTCAATATATATATATATATATATATATATCTT\n",
            "ATTATATAAAGTATATATATAAGTTGAAATTTATTTTTCATTCATATCAATTAAATTTCA\n",
            "ATTATTCTTAATGAATTTTCTAATTACACAACTATCTCATAACCCCAAACTAAAAAATGT\n",
            "GATGTATGAGATAGTTTATCCTAGCACAAATACTTCTGATTAGTTTCACTGTTCAAACCA\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!hisat2-build --version"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "7q7rOfsngEyu",
        "outputId": "fcd4536e-d64e-48ed-e94f-b570456d97ca"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "hisat2-build version 2.2.1\n",
            "64-bit\n",
            "Built on fv-az212-765\n",
            "Wed Feb 23 05:51:07 UTC 2022\n",
            "Compiler: collect2: error: ld returned 1 exit status\n",
            "Options: -O3 -m64 -msse2 -funroll-loops -g3 -DPOPCNT_CAPABILITY -std=c++11\n",
            "Sizeof {int, long, long long, void*, size_t, off_t}: {4, 8, 8, 8, 8, 8}\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "#git setion"
      ],
      "metadata": {
        "id": "h8iY2eIvF9h_"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!git clone https://github.com/AliMohseni05/rna-seq-snakemake.git"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "YXuPA9qm7CbM",
        "outputId": "55048739-ca15-45bf-ffe3-f2645bc927df"
      },
      "execution_count": 32,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Cloning into 'rna-seq-snakemake'...\n",
            "remote: Enumerating objects: 3, done.\u001b[K\n",
            "remote: Counting objects: 100% (3/3), done.\u001b[K\n",
            "remote: Total 3 (delta 0), reused 0 (delta 0), pack-reused 0 (from 0)\u001b[K\n",
            "Receiving objects: 100% (3/3), done.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!git config --global user.email \"skysly42@gmail.com\"\n",
        "!git config --global user.name \"colab sky\"\n"
      ],
      "metadata": {
        "id": "YI2zvGykHFi3"
      },
      "execution_count": 41,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "%cd rna-seq-snakemake"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "jpx2rCYTG0eG",
        "outputId": "1ea9cd6b-99c0-4502-9eb2-f9150471f047"
      },
      "execution_count": 39,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/content/rna-seq-snakemake\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!git add .\n",
        "!git commit -m \"Update from Colab\""
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ZxAaapW7F4EI",
        "outputId": "e72ff7a4-e825-40b5-b892-08b47e0fa1b1"
      },
      "execution_count": 42,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[main 6454088] Update from Colab\n",
            " 2 files changed, 176 insertions(+)\n",
            " create mode 100644 Snakefile\n",
            " create mode 100644 config.yaml\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!git remote add upstream https://github.com/AliMohseni05/rna-seq-snakemake.git\n"
      ],
      "metadata": {
        "id": "MsOmMUrRF6eU"
      },
      "execution_count": 43,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!git remote -v"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "tv5LahIlHgCb",
        "outputId": "aaeb5826-85a9-42e7-c5c5-dc7d5e9416e7"
      },
      "execution_count": 45,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "origin\thttps://github.com/AliMohseni05/rna-seq-snakemake.git (fetch)\n",
            "origin\thttps://github.com/AliMohseni05/rna-seq-snakemake.git (push)\n",
            "upstream\thttps://github.com/AliMohseni05/rna-seq-snakemake.git (fetch)\n",
            "upstream\thttps://github.com/AliMohseni05/rna-seq-snakemake.git (push)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!git push https://USERNAME:TOKEN@github.com/AliMohseni05/rna-seq-snakemake.git\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "-F6wAXw4HYlp",
        "outputId": "82f41248-678e-48d9-bd27-2d5846f497a7"
      },
      "execution_count": 46,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "fatal: could not read Username for 'https://github.com': No such device or address\n"
          ]
        }
      ]
    }
  ]
}